In [1]:
# firsttime = true

if !(@isdefined firsttime) || firsttime == true
    include("HMAlg.jl")
    firsttime = false
end
using Main.HMAlg
# using .HMAlg

using Symbolics
using Combinatorics

In [2]:
test = HM(6, 4, 1, symbol=:a)

HM(a[1:6,1:4,1:1])

In [3]:
test.Hx

a[1:6,1:4,1:1]

In [4]:
list_fromsymbol(test.Hx[1,1,1])

3-element Vector{Any}:
 1
 1
 1

In [5]:
test_static = genStaticHM(test.Hx)

general method called! a[1:6,1:4,1:1]


6×4×1 Array{Num, 3}:
[:, :, 1] =
 a[1, 1, 1]  a[1, 2, 1]  a[1, 3, 1]  a[1, 4, 1]
 a[2, 1, 1]  a[2, 2, 1]  a[2, 3, 1]  a[2, 4, 1]
 a[3, 1, 1]  a[3, 2, 1]  a[3, 3, 1]  a[3, 4, 1]
 a[4, 1, 1]  a[4, 2, 1]  a[4, 3, 1]  a[4, 4, 1]
 a[5, 1, 1]  a[5, 2, 1]  a[5, 3, 1]  a[5, 4, 1]
 a[6, 1, 1]  a[6, 2, 1]  a[6, 3, 1]  a[6, 4, 1]

In [6]:
view_fromsymbol(test_static[1,1])[3] = 1

1

In [7]:
test_static

6×4×1 Array{Num, 3}:
[:, :, 1] =
 a[1, 1, 1]  a[1, 2, 1]  a[1, 3, 1]  a[1, 4, 1]
 a[2, 1, 1]  a[2, 2, 1]  a[2, 3, 1]  a[2, 4, 1]
 a[3, 1, 1]  a[3, 2, 1]  a[3, 3, 1]  a[3, 4, 1]
 a[4, 1, 1]  a[4, 2, 1]  a[4, 3, 1]  a[4, 4, 1]
 a[5, 1, 1]  a[5, 2, 1]  a[5, 3, 1]  a[5, 4, 1]
 a[6, 1, 1]  a[6, 2, 1]  a[6, 3, 1]  a[6, 4, 1]

In [8]:
simplify(test_static[1,1]*test_static[1,2])

a[1, 1, 1]*a[1, 2, 1]

In [9]:
simplify(test_static[1,1]*test_static[1,2] + test.Hx[1,1]*test.Hx[1,2])

2a[1, 1, 1]*a[1, 2, 1]

In [10]:
simplify(test_static[1,1]*test_static[1,2] + test.Hx[1,2]*test.Hx[1,2])

a[1, 1, 1]*a[1, 2, 1] + a[1, 2, 1]^2

In [11]:
simplify(test_static[1,1]*test_static[1,2] + test.Hx[1,3]*test.Hx[1,2])

a[1, 1, 1]*a[1, 2, 1] + a[1, 2, 1]*a[1, 3, 1]

In [14]:
fam3 = HM(repeat([4],6)..., symbol=:b)

HM(b[1:4,1:4,1:4,1:4,1:4,1:4])

In [15]:
permutations(1:5)

Combinatorics.Permutations{UnitRange{Int64}}(1:5, 5)

In [22]:
orbit = [HMAlg.compose(σ4, [2,3,3,4,4,4], σ6) for σ4 in permutations(1:4) for σ6 in permutations(1:6)]

17280-element Vector{Vector{Int64}}:
 [2, 3, 3, 4, 4, 4]
 [2, 3, 3, 4, 4, 4]
 [2, 3, 3, 4, 4, 4]
 [2, 3, 3, 4, 4, 4]
 [2, 3, 3, 4, 4, 4]
 [2, 3, 3, 4, 4, 4]
 [2, 3, 4, 3, 4, 4]
 [2, 3, 4, 3, 4, 4]
 [2, 3, 4, 4, 3, 4]
 [2, 3, 4, 4, 4, 3]
 [2, 3, 4, 4, 3, 4]
 [2, 3, 4, 4, 4, 3]
 [2, 3, 4, 3, 4, 4]
 ⋮
 [1, 1, 2, 3, 2, 1]
 [1, 1, 2, 3, 1, 2]
 [1, 1, 2, 2, 3, 1]
 [1, 1, 2, 2, 1, 3]
 [1, 1, 2, 1, 3, 2]
 [1, 1, 2, 1, 2, 3]
 [1, 1, 1, 3, 2, 2]
 [1, 1, 1, 3, 2, 2]
 [1, 1, 1, 2, 3, 2]
 [1, 1, 1, 2, 2, 3]
 [1, 1, 1, 2, 3, 2]
 [1, 1, 1, 2, 2, 3]

In [23]:
unique_orbit = unique(orbit)

1440-element Vector{Vector{Int64}}:
 [2, 3, 3, 4, 4, 4]
 [2, 3, 4, 3, 4, 4]
 [2, 3, 4, 4, 3, 4]
 [2, 3, 4, 4, 4, 3]
 [2, 4, 3, 3, 4, 4]
 [2, 4, 3, 4, 3, 4]
 [2, 4, 3, 4, 4, 3]
 [2, 4, 4, 3, 3, 4]
 [2, 4, 4, 3, 4, 3]
 [2, 4, 4, 4, 3, 3]
 [3, 2, 3, 4, 4, 4]
 [3, 2, 4, 3, 4, 4]
 [3, 2, 4, 4, 3, 4]
 ⋮
 [1, 1, 3, 2, 2, 1]
 [1, 1, 3, 2, 1, 2]
 [1, 1, 3, 1, 2, 2]
 [1, 1, 2, 3, 2, 1]
 [1, 1, 2, 3, 1, 2]
 [1, 1, 2, 2, 3, 1]
 [1, 1, 2, 2, 1, 3]
 [1, 1, 2, 1, 3, 2]
 [1, 1, 2, 1, 2, 3]
 [1, 1, 1, 3, 2, 2]
 [1, 1, 1, 2, 3, 2]
 [1, 1, 1, 2, 2, 3]

In [28]:
orbit2 = [HMAlg.compose(σ4, [1,2,3,3,4,4], σ6) for σ4 in permutations(1:4) for σ6 in permutations(1:6)]
unique_orbit2 = unique(orbit2)

1080-element Vector{Vector{Int64}}:
 [1, 2, 3, 3, 4, 4]
 [1, 2, 3, 4, 3, 4]
 [1, 2, 3, 4, 4, 3]
 [1, 2, 4, 3, 3, 4]
 [1, 2, 4, 3, 4, 3]
 [1, 2, 4, 4, 3, 3]
 [1, 3, 2, 3, 4, 4]
 [1, 3, 2, 4, 3, 4]
 [1, 3, 2, 4, 4, 3]
 [1, 3, 3, 2, 4, 4]
 [1, 3, 3, 4, 2, 4]
 [1, 3, 3, 4, 4, 2]
 [1, 3, 4, 2, 3, 4]
 ⋮
 [2, 2, 3, 4, 1, 1]
 [2, 2, 3, 1, 4, 1]
 [2, 2, 3, 1, 1, 4]
 [2, 2, 4, 3, 1, 1]
 [2, 2, 4, 1, 3, 1]
 [2, 2, 4, 1, 1, 3]
 [2, 2, 1, 3, 4, 1]
 [2, 2, 1, 3, 1, 4]
 [2, 2, 1, 4, 3, 1]
 [2, 2, 1, 4, 1, 3]
 [2, 2, 1, 1, 3, 4]
 [2, 2, 1, 1, 4, 3]

In [108]:
orbit3 = [HMAlg.compose.([σ4], [[1,2,3,3,4,4],[1,1,3,1,2,2]], [σ6]) for σ4 in permutations(1:4) for σ6 in permutations(1:6)]
unique_orbit3 = unique(Set(x) for x in orbit3)

8640-element Vector{Set{Vector{Int64}}}:
 Set([[1, 1, 3, 1, 2, 2], [1, 2, 3, 3, 4, 4]])
 Set([[1, 1, 3, 2, 1, 2], [1, 2, 3, 4, 3, 4]])
 Set([[1, 1, 3, 2, 2, 1], [1, 2, 3, 4, 4, 3]])
 Set([[1, 1, 1, 3, 2, 2], [1, 2, 3, 3, 4, 4]])
 Set([[1, 1, 1, 2, 3, 2], [1, 2, 3, 4, 3, 4]])
 Set([[1, 2, 3, 4, 4, 3], [1, 1, 1, 2, 2, 3]])
 Set([[1, 1, 2, 3, 1, 2], [1, 2, 4, 3, 3, 4]])
 Set([[1, 2, 4, 3, 4, 3], [1, 1, 2, 3, 2, 1]])
 Set([[1, 1, 2, 1, 3, 2], [1, 2, 4, 3, 3, 4]])
 Set([[1, 1, 2, 1, 2, 3], [1, 2, 4, 3, 4, 3]])
 Set([[1, 1, 2, 2, 3, 1], [1, 2, 4, 4, 3, 3]])
 Set([[1, 2, 4, 4, 3, 3], [1, 1, 2, 2, 1, 3]])
 Set([[1, 3, 2, 3, 4, 4], [1, 3, 1, 1, 2, 2]])
 ⋮
 Set([[3, 3, 2, 4, 4, 4], [1, 1, 2, 4, 3, 2]])
 Set([[3, 3, 2, 4, 4, 4], [1, 1, 2, 4, 2, 3]])
 Set([[3, 3, 2, 4, 4, 4], [1, 1, 2, 3, 4, 2]])
 Set([[3, 3, 2, 4, 4, 4], [1, 1, 2, 3, 2, 4]])
 Set([[3, 3, 2, 4, 4, 4], [1, 1, 2, 2, 4, 3]])
 Set([[3, 3, 2, 4, 4, 4], [1, 1, 2, 2, 3, 4]])
 Set([[3, 3, 4, 4, 4, 2], [1, 1, 2, 4, 3, 2]])
 Set([[1, 1, 2, 

In [43]:
#Testing 
indexable_function(fi)::Function = x -> fi[x]

eager_compose(f1, f2) = map(indexable_function(f1),f2)

(compose(f1::Tuple{Vararg{T}}, f2::Tuple{Vararg{T}})::Tuple{Vararg{T}}) where {T} = eager_compose(f1, f2)

(∘(f1::Tuple{Vararg{T}}, f2::Tuple{Vararg{T}})::Tuple{Vararg{T}}) where {T} = compose(f1, f2)

In [165]:
remaining_set = Set(collect(x) for x in Iterators.product(repeat([1:4],6)...))

Set{Vector{Int64}} with 4096 elements:
  [1, 4, 4, 4, 3, 2]
  [1, 1, 1, 1, 4, 1]
  [3, 3, 2, 1, 4, 1]
  [1, 3, 3, 2, 4, 1]
  [2, 2, 1, 1, 4, 3]
  [1, 2, 3, 3, 1, 4]
  [4, 2, 3, 4, 1, 4]
  [3, 4, 2, 2, 2, 4]
  [3, 4, 4, 1, 3, 4]
  [4, 2, 2, 1, 4, 1]
  [3, 2, 4, 1, 2, 1]
  [1, 3, 2, 4, 3, 4]
  [1, 3, 2, 1, 2, 2]
  [4, 1, 1, 3, 1, 4]
  [2, 4, 4, 3, 2, 3]
  [2, 1, 4, 4, 4, 2]
  [3, 4, 4, 4, 4, 1]
  [1, 1, 2, 4, 1, 1]
  [2, 2, 2, 2, 2, 2]
  [4, 1, 3, 2, 2, 2]
  [2, 4, 1, 3, 2, 2]
  [2, 2, 4, 1, 3, 2]
  [2, 2, 2, 4, 1, 3]
  [1, 1, 3, 3, 1, 3]
  [4, 1, 3, 4, 1, 3]
  ⋮ 

In [166]:
count = 0
groups1 = []
while !isempty(remaining_set)
    cur_f   = pop!(remaining_set)
    cur_set = Set(HMAlg.compose(σ4, cur_f, σ6) for σ4 in permutations(1:4) for σ6 in permutations(1:6))
    push!(groups1, cur_set)
#     if cur_set ⊆ (remaining_set ∪ Set([cur_f]))
#         println("Yay!")
#     else
#         println("Oops!")
#     end
    setdiff!(remaining_set, cur_set)
    count += 1
end

count

9

In [130]:
groups1

9-element Vector{Any}:
 Set([[1, 4, 4, 4, 3, 2], [4, 2, 3, 4, 1, 4], [4, 2, 3, 1, 3, 3], [2, 2, 3, 1, 4, 2], [4, 2, 1, 2, 3, 2], [3, 4, 4, 2, 1, 4], [4, 2, 3, 2, 2, 1], [2, 1, 2, 4, 3, 2], [1, 4, 4, 3, 4, 2], [2, 3, 3, 4, 3, 1]  …  [1, 4, 3, 4, 4, 2], [2, 4, 2, 1, 3, 2], [1, 3, 2, 4, 1, 1], [1, 1, 4, 3, 1, 2], [2, 2, 2, 4, 3, 1], [4, 4, 2, 3, 1, 4], [1, 3, 3, 4, 2, 3], [2, 4, 3, 3, 1, 3], [4, 4, 4, 2, 1, 3], [2, 2, 2, 3, 4, 1]])
 Set([[4, 4, 4, 4, 2, 4], [1, 1, 1, 2, 1, 1], [1, 1, 1, 1, 4, 1], [4, 4, 4, 4, 4, 2], [2, 2, 3, 2, 2, 2], [3, 4, 3, 3, 3, 3], [3, 3, 1, 3, 3, 3], [3, 3, 3, 3, 3, 2], [2, 2, 2, 2, 1, 2], [4, 4, 4, 2, 4, 4]  …  [2, 2, 2, 2, 2, 4], [1, 3, 1, 1, 1, 1], [2, 2, 2, 2, 3, 2], [3, 1, 1, 1, 1, 1], [2, 2, 2, 2, 4, 2], [4, 4, 4, 4, 1, 4], [2, 4, 2, 2, 2, 2], [3, 3, 3, 1, 3, 3], [4, 2, 4, 4, 4, 4], [1, 1, 1, 1, 3, 1]])
 Set([[3, 3, 2, 1, 4, 1], [1, 3, 3, 2, 4, 1], [2, 2, 1, 1, 4, 3], [1, 2, 3, 3, 1, 4], [4, 2, 3, 2, 1, 3], [3, 4, 2, 4, 3, 1], [2, 1, 3, 2, 3, 4], [3, 2, 4, 1

In [169]:
gen_new_6_to_3 = () -> (HMAlg.compose(σ3, [1,2,2,3,3,3], σ6) for σ3 in permutations(1:3) for σ6 in permutations(1:6))
remaining_set = Set(Set(collect(x)) for x in Iterators.product(gen_new_6_to_3(), gen_new_6_to_3()))

Set{Set{Vector{Int64}}} with 64980 elements:
  Set([[2, 2, 3, 1, 1, 1], [1, 1, 3, 2, 1, 3]])
  Set([[2, 1, 3, 1, 3, 3], [3, 3, 3, 2, 2, 1]])
  Set([[2, 3, 2, 1, 1, 2], [2, 2, 1, 3, 3, 2]])
  Set([[1, 1, 3, 1, 2, 2], [3, 3, 3, 2, 2, 1]])
  Set([[3, 3, 2, 2, 1, 3], [1, 1, 2, 3, 1, 3]])
  Set([[1, 3, 3, 2, 1, 3], [3, 1, 3, 3, 2, 1]])
  Set([[3, 3, 2, 2, 1, 3], [3, 3, 2, 1, 3, 2]])
  Set([[2, 2, 3, 1, 2, 1], [3, 1, 1, 3, 1, 2]])
  Set([[2, 3, 3, 2, 3, 1], [3, 2, 2, 3, 1, 2]])
  Set([[1, 1, 2, 3, 3, 3], [2, 2, 1, 1, 3, 1]])
  Set([[2, 1, 1, 1, 3, 3], [1, 2, 3, 1, 3, 1]])
  Set([[3, 3, 1, 1, 2, 1], [1, 2, 2, 3, 3, 2]])
  Set([[1, 1, 3, 2, 1, 2], [1, 3, 3, 1, 3, 2]])
  Set([[2, 3, 2, 1, 2, 1], [1, 1, 2, 2, 3, 2]])
  Set([[1, 3, 1, 2, 3, 1], [2, 1, 3, 3, 1, 3]])
  Set([[3, 1, 1, 3, 2, 3], [2, 3, 2, 1, 3, 2]])
  Set([[1, 3, 1, 2, 2, 2], [1, 2, 1, 2, 1, 3]])
  Set([[2, 3, 3, 3, 1, 1], [1, 1, 3, 2, 2, 1]])
  Set([[2, 1, 2, 2, 3, 1], [3, 3, 1, 2, 3, 2]])
  Set([[1, 3, 2, 1, 1, 3], [1, 3, 3, 3, 2, 

In [171]:
count = 0
groups2 = []
while !isempty(remaining_set)
    cur_f   = pop!(remaining_set)
    cur_set = Set(Set(HMAlg.compose.([σ3], cur_f, [σ6])) for σ3 in permutations(1:3) for σ6 in permutations(1:6))
    push!(groups2, cur_set)
#     if cur_set ⊆ (remaining_set ∪ Set([cur_f]))
#         println("Yay!")
#     else
#         println("Oops!")
#     end
    setdiff!(remaining_set, cur_set)
    count += 1
end

count

52

In [173]:
foreach(x -> display(x), groups2)

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[2, 2, 3, 1, 1, 1], [1, 1, 3, 2, 1, 3]])
  Set([[2, 1, 1, 3, 2, 2], [3, 2, 2, 3, 2, 1]])
  Set([[1, 2, 1, 3, 2, 1], [1, 2, 3, 1, 1, 3]])
  Set([[2, 2, 2, 3, 1, 3], [3, 1, 2, 2, 1, 2]])
  Set([[3, 3, 3, 2, 1, 2], [1, 3, 2, 3, 1, 3]])
  Set([[1, 1, 3, 3, 3, 2], [3, 3, 3, 1, 2, 2]])
  Set([[2, 2, 2, 1, 3, 1], [2, 3, 3, 2, 2, 1]])
  Set([[3, 3, 2, 3, 2, 1], [2, 1, 3, 3, 3, 1]])
  Set([[1, 2, 3, 3, 2, 2], [2, 1, 2, 3, 2, 1]])
  Set([[3, 2, 3, 1, 2, 2], [3, 1, 2, 2, 1, 2]])
  Set([[2, 2, 3, 1, 2, 1], [3, 3, 2, 2, 2, 1]])
  Set([[3, 2, 1, 2, 1, 2], [3, 3, 2, 2, 2, 1]])
  Set([[2, 1, 2, 1, 3, 2], [1, 2, 3, 2, 3, 2]])
  Set([[1, 1, 2, 3, 2, 2], [1, 2, 3, 2, 2, 3]])
  Set([[3, 2, 3, 2, 1, 3], [1, 3, 3, 2, 3, 1]])
  Set([[2, 3, 3, 2, 2, 1], [1, 2, 2, 2, 3, 1]])
  Set([[2, 1, 3, 1, 3, 3], [2, 3, 1, 3, 2, 3]])
  Set([[3, 1, 2, 3, 3, 1], [2, 3, 3, 2, 3, 1]])
  Set([[2, 3, 1, 2, 2, 1], [2, 2, 1, 3, 3, 2]])
  Set([[1, 3, 2, 3, 3, 1], [1, 2, 3, 2, 3, 3

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[3, 2, 1, 1, 3, 1], [1, 1, 3, 2, 2, 1]])
  Set([[2, 1, 3, 1, 3, 3], [3, 3, 3, 2, 2, 1]])
  Set([[2, 2, 3, 1, 2, 1], [3, 1, 2, 2, 2, 3]])
  Set([[1, 2, 3, 1, 1, 2], [3, 3, 1, 2, 1, 1]])
  Set([[1, 2, 3, 1, 1, 2], [3, 3, 1, 1, 2, 1]])
  Set([[3, 1, 2, 2, 2, 1], [2, 3, 3, 1, 2, 2]])
  Set([[3, 2, 1, 3, 2, 3], [3, 3, 3, 2, 1, 1]])
  Set([[2, 1, 2, 2, 1, 3], [3, 3, 2, 1, 2, 2]])
  Set([[2, 2, 3, 1, 3, 3], [1, 3, 2, 3, 1, 3]])
  Set([[3, 1, 3, 1, 2, 1], [2, 3, 1, 2, 1, 1]])
  Set([[1, 1, 2, 2, 3, 2], [3, 2, 2, 1, 2, 3]])
  Set([[3, 2, 2, 1, 1, 2], [2, 2, 3, 3, 2, 1]])
  Set([[3, 1, 3, 2, 2, 3], [1, 3, 3, 1, 3, 2]])
  Set([[1, 2, 1, 1, 3, 3], [2, 1, 3, 1, 1, 2]])
  Set([[1, 3, 1, 2, 2, 1], [1, 1, 3, 1, 3, 2]])
  Set([[2, 3, 2, 1, 2, 3], [2, 1, 1, 2, 3, 2]])
  Set([[2, 2, 3, 3, 3, 1], [1, 3, 2, 3, 1, 3]])
  Set([[1, 1, 3, 3, 2, 1], [2, 3, 1, 2, 1, 1]])
  Set([[3, 3, 1, 2, 2, 3], [3, 1, 3, 3, 1, 2]])
  Set([[3, 2, 1, 2, 2, 3], [2, 1, 2, 3, 2, 1

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[2, 1, 3, 2, 2, 1], [1, 3, 2, 2, 2, 3]])
  Set([[2, 3, 3, 3, 1, 1], [3, 3, 3, 1, 2, 2]])
  Set([[3, 1, 2, 2, 1, 1], [1, 1, 3, 3, 1, 2]])
  Set([[2, 3, 2, 1, 1, 2], [2, 2, 1, 3, 3, 2]])
  Set([[3, 1, 3, 3, 2, 2], [3, 3, 3, 2, 1, 1]])
  Set([[2, 2, 3, 3, 1, 2], [3, 2, 1, 1, 2, 2]])
  Set([[1, 1, 2, 1, 3, 2], [2, 1, 3, 1, 1, 3]])
  Set([[1, 1, 3, 1, 2, 2], [2, 1, 1, 1, 3, 3]])
  Set([[3, 2, 2, 1, 1, 2], [2, 2, 1, 3, 3, 2]])
  Set([[3, 3, 1, 2, 1, 1], [2, 2, 1, 1, 1, 3]])
  Set([[2, 3, 1, 1, 2, 1], [3, 1, 1, 2, 3, 1]])
  Set([[1, 1, 2, 1, 3, 2], [1, 2, 3, 1, 1, 3]])
  Set([[2, 3, 3, 3, 2, 1], [1, 3, 3, 2, 1, 3]])
  Set([[1, 3, 1, 2, 1, 2], [1, 1, 2, 3, 1, 3]])
  Set([[2, 3, 2, 1, 1, 1], [3, 1, 3, 1, 1, 2]])
  Set([[2, 3, 1, 1, 2, 1], [3, 1, 2, 1, 3, 1]])
  Set([[1, 2, 3, 3, 2, 3], [3, 1, 3, 2, 1, 3]])
  Set([[1, 3, 2, 2, 2, 3], [2, 1, 2, 3, 2, 1]])
  Set([[3, 2, 2, 1, 2, 1], [2, 1, 2, 3, 2, 3]])
  Set([[2, 3, 2, 1, 1, 2], [1, 2, 2, 3, 3, 2]

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[1, 3, 1, 2, 1, 2], [3, 3, 3, 1, 2, 2]])
  Set([[2, 2, 1, 1, 2, 3], [3, 2, 1, 3, 3, 1]])
  Set([[3, 1, 1, 3, 1, 2], [3, 2, 2, 1, 3, 2]])
  Set([[3, 3, 2, 3, 2, 1], [2, 1, 2, 1, 3, 1]])
  Set([[2, 3, 1, 2, 3, 3], [3, 1, 1, 2, 1, 2]])
  Set([[1, 3, 2, 2, 1, 2], [1, 1, 3, 3, 3, 2]])
  Set([[1, 1, 3, 3, 2, 3], [1, 3, 2, 2, 2, 1]])
  Set([[2, 3, 1, 2, 2, 1], [3, 3, 2, 1, 3, 1]])
  Set([[3, 1, 1, 2, 3, 1], [3, 3, 2, 2, 1, 2]])
  Set([[1, 1, 3, 1, 2, 2], [3, 3, 3, 2, 2, 1]])
  Set([[3, 2, 1, 1, 3, 1], [1, 2, 2, 2, 3, 3]])
  Set([[3, 3, 3, 1, 2, 1], [3, 2, 2, 2, 1, 1]])
  Set([[3, 2, 3, 1, 2, 2], [3, 3, 2, 1, 1, 1]])
  Set([[1, 2, 3, 2, 1, 1], [3, 2, 3, 1, 3, 2]])
  Set([[1, 2, 1, 1, 3, 3], [3, 2, 2, 2, 3, 1]])
  Set([[2, 2, 1, 3, 1, 2], [3, 2, 3, 1, 1, 3]])
  Set([[2, 1, 3, 1, 3, 3], [1, 1, 2, 2, 3, 2]])
  Set([[3, 1, 2, 1, 1, 2], [2, 3, 3, 1, 3, 2]])
  Set([[2, 2, 3, 3, 1, 3], [3, 2, 2, 1, 1, 1]])
  Set([[3, 1, 2, 1, 2, 2], [1, 1, 3, 3, 2, 3

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[3, 1, 1, 3, 2, 3], [3, 3, 1, 2, 2, 2]])
  Set([[2, 3, 3, 1, 1, 3], [2, 1, 1, 1, 2, 3]])
  Set([[1, 2, 3, 3, 1, 3], [2, 2, 1, 3, 1, 1]])
  Set([[3, 1, 2, 1, 1, 3], [3, 1, 2, 3, 3, 2]])
  Set([[2, 3, 1, 2, 3, 3], [2, 3, 1, 3, 1, 1]])
  Set([[2, 1, 2, 1, 2, 3], [2, 3, 1, 1, 1, 3]])
  Set([[1, 1, 1, 3, 2, 2], [2, 2, 1, 3, 3, 2]])
  Set([[3, 1, 1, 2, 2, 2], [3, 1, 3, 1, 1, 2]])
  Set([[1, 2, 1, 1, 3, 3], [1, 2, 2, 2, 3, 1]])
  Set([[3, 1, 3, 3, 2, 2], [3, 1, 2, 2, 1, 2]])
  Set([[2, 1, 2, 1, 2, 3], [3, 2, 3, 1, 2, 3]])
  Set([[1, 1, 2, 3, 2, 1], [1, 3, 2, 3, 1, 3]])
  Set([[3, 3, 2, 2, 1, 3], [1, 1, 2, 3, 1, 3]])
  Set([[3, 3, 2, 2, 1, 2], [1, 3, 3, 2, 1, 3]])
  Set([[3, 2, 1, 3, 1, 1], [1, 2, 2, 3, 1, 2]])
  Set([[1, 3, 2, 2, 3, 2], [1, 2, 1, 1, 3, 2]])
  Set([[2, 1, 1, 3, 3, 3], [2, 3, 1, 2, 3, 2]])
  Set([[2, 3, 2, 3, 1, 2], [1, 2, 1, 3, 1, 2]])
  Set([[3, 2, 2, 1, 3, 2], [1, 2, 3, 1, 3, 3]])
  Set([[1, 2, 3, 3, 2, 3], [1, 2, 1, 1, 3, 3

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[1, 2, 3, 2, 2, 1], [3, 1, 2, 1, 2, 2]])
  Set([[1, 3, 1, 3, 1, 2], [1, 2, 3, 1, 3, 1]])
  Set([[2, 3, 3, 1, 1, 1], [3, 1, 1, 3, 1, 2]])
  Set([[1, 3, 3, 2, 2, 2], [3, 2, 2, 3, 1, 2]])
  Set([[3, 3, 2, 1, 1, 1], [1, 2, 1, 3, 3, 1]])
  Set([[3, 3, 1, 2, 1, 3], [1, 1, 2, 3, 3, 3]])
  Set([[3, 2, 1, 3, 3, 2], [1, 3, 2, 3, 2, 3]])
  Set([[1, 2, 2, 3, 2, 3], [2, 3, 3, 1, 2, 2]])
  Set([[2, 2, 3, 1, 3, 3], [3, 1, 2, 3, 3, 2]])
  Set([[1, 3, 2, 1, 3, 1], [3, 1, 1, 1, 2, 3]])
  Set([[3, 3, 2, 1, 3, 2], [2, 1, 3, 2, 3, 3]])
  Set([[1, 2, 3, 1, 2, 1], [2, 3, 1, 1, 1, 2]])
  Set([[3, 1, 1, 1, 3, 2], [1, 2, 3, 1, 1, 3]])
  Set([[3, 1, 3, 3, 1, 2], [3, 2, 1, 1, 3, 3]])
  Set([[1, 1, 3, 3, 1, 2], [1, 3, 1, 1, 2, 3]])
  Set([[2, 1, 3, 3, 1, 3], [1, 3, 3, 1, 3, 2]])
  Set([[1, 3, 1, 1, 2, 3], [3, 2, 1, 3, 1, 1]])
  Set([[2, 2, 2, 3, 1, 1], [1, 1, 2, 2, 2, 3]])
  Set([[1, 1, 2, 2, 3, 2], [2, 3, 1, 2, 2, 1]])
  Set([[3, 1, 2, 2, 2, 1], [1, 2, 1, 3, 2, 2

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[3, 1, 1, 2, 2, 1], [1, 2, 1, 2, 3, 1]])
  Set([[1, 3, 2, 1, 3, 3], [1, 3, 3, 2, 3, 1]])
  Set([[3, 1, 3, 2, 3, 1], [3, 3, 3, 1, 2, 1]])
  Set([[1, 2, 3, 3, 2, 3], [3, 1, 3, 2, 2, 3]])
  Set([[1, 3, 3, 1, 2, 3], [1, 3, 1, 2, 3, 3]])
  Set([[1, 3, 2, 1, 1, 2], [1, 1, 2, 2, 1, 3]])
  Set([[3, 3, 1, 2, 1, 1], [2, 3, 3, 1, 1, 1]])
  Set([[2, 1, 3, 1, 1, 3], [1, 3, 2, 1, 1, 3]])
  Set([[2, 1, 2, 1, 3, 2], [2, 1, 2, 2, 1, 3]])
  Set([[1, 1, 2, 3, 2, 2], [3, 1, 2, 2, 1, 2]])
  Set([[2, 3, 2, 3, 3, 1], [2, 3, 3, 1, 3, 2]])
  Set([[2, 2, 3, 1, 2, 3], [2, 1, 2, 3, 2, 3]])
  Set([[1, 1, 2, 3, 2, 2], [1, 2, 2, 1, 3, 2]])
  Set([[3, 2, 2, 2, 1, 1], [1, 2, 2, 3, 1, 2]])
  Set([[3, 1, 1, 1, 3, 2], [3, 3, 1, 1, 2, 1]])
  Set([[3, 1, 1, 3, 2, 1], [3, 1, 1, 2, 1, 3]])
  Set([[2, 1, 1, 3, 2, 2], [3, 2, 1, 1, 2, 2]])
  Set([[3, 2, 1, 3, 2, 2], [2, 1, 3, 3, 2, 2]])
  Set([[2, 3, 3, 1, 2, 2], [3, 3, 1, 2, 2, 2]])
  Set([[2, 1, 1, 1, 3, 3], [1, 1, 3, 1, 3, 2

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[3, 3, 2, 1, 3, 1], [1, 2, 3, 2, 2, 3]])
  Set([[1, 2, 1, 3, 3, 3], [2, 1, 3, 2, 1, 1]])
  Set([[3, 1, 3, 1, 3, 2], [1, 2, 1, 3, 2, 1]])
  Set([[2, 2, 1, 3, 2, 1], [1, 3, 2, 1, 1, 3]])
  Set([[3, 2, 3, 2, 3, 1], [2, 1, 1, 3, 1, 3]])
  Set([[1, 2, 3, 3, 1, 1], [2, 1, 2, 1, 2, 3]])
  Set([[2, 1, 2, 2, 3, 1], [3, 3, 1, 3, 2, 2]])
  Set([[1, 2, 3, 3, 3, 2], [2, 1, 1, 2, 2, 3]])
  Set([[2, 1, 2, 1, 3, 2], [3, 2, 3, 3, 2, 1]])
  Set([[1, 3, 2, 3, 3, 1], [3, 1, 1, 1, 2, 2]])
  Set([[2, 1, 3, 3, 3, 2], [1, 3, 2, 1, 1, 3]])
  Set([[1, 2, 3, 3, 2, 2], [2, 1, 1, 2, 1, 3]])
  Set([[1, 2, 2, 3, 2, 1], [3, 1, 1, 1, 3, 2]])
  Set([[3, 2, 3, 3, 2, 1], [2, 3, 1, 2, 1, 2]])
  Set([[3, 1, 2, 2, 1, 2], [1, 3, 1, 3, 2, 1]])
  Set([[2, 2, 3, 1, 3, 2], [1, 3, 2, 2, 1, 1]])
  Set([[3, 2, 3, 1, 3, 2], [2, 3, 2, 2, 1, 1]])
  Set([[3, 1, 1, 2, 1, 3], [2, 2, 3, 3, 3, 1]])
  Set([[1, 3, 3, 3, 2, 2], [3, 1, 1, 2, 3, 1]])
  Set([[2, 1, 1, 2, 3, 1], [3, 3, 2, 1, 1, 3

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[1, 3, 1, 3, 2, 3], [3, 2, 3, 1, 1, 1]])
  Set([[2, 1, 3, 1, 3, 3], [1, 3, 1, 3, 1, 2]])
  Set([[2, 3, 1, 3, 3, 2], [3, 1, 2, 2, 2, 3]])
  Set([[2, 2, 1, 2, 1, 3], [1, 1, 2, 3, 2, 1]])
  Set([[3, 2, 2, 1, 1, 2], [1, 3, 1, 2, 2, 1]])
  Set([[2, 1, 2, 3, 3, 3], [3, 2, 3, 2, 1, 2]])
  Set([[1, 1, 2, 3, 3, 3], [3, 3, 1, 1, 1, 2]])
  Set([[2, 1, 1, 1, 2, 3], [1, 2, 3, 2, 1, 2]])
  Set([[3, 1, 2, 1, 2, 2], [1, 2, 3, 2, 1, 1]])
  Set([[2, 2, 3, 3, 1, 3], [3, 3, 2, 2, 2, 1]])
  Set([[2, 3, 3, 2, 3, 1], [3, 2, 2, 3, 1, 2]])
  Set([[1, 1, 3, 2, 2, 2], [2, 2, 1, 3, 1, 1]])
  Set([[1, 2, 1, 1, 3, 3], [2, 3, 3, 3, 1, 1]])
  Set([[2, 3, 3, 3, 2, 1], [3, 2, 2, 1, 3, 2]])
  Set([[1, 3, 1, 2, 3, 1], [2, 1, 3, 3, 1, 3]])
  Set([[3, 3, 2, 1, 1, 3], [1, 2, 1, 3, 3, 1]])
  Set([[3, 3, 1, 3, 2, 1], [2, 1, 3, 1, 1, 3]])
  Set([[3, 1, 3, 2, 2, 3], [2, 2, 1, 3, 3, 2]])
  Set([[3, 3, 1, 2, 1, 3], [1, 1, 3, 1, 3, 2]])
  Set([[2, 3, 3, 1, 3, 2], [3, 1, 2, 2, 2, 3]

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[2, 3, 2, 1, 1, 1], [1, 3, 1, 3, 3, 2]])
  Set([[3, 3, 2, 1, 1, 1], [2, 2, 3, 1, 3, 3]])
  Set([[2, 1, 1, 2, 3, 1], [1, 3, 2, 1, 3, 3]])
  Set([[1, 2, 2, 3, 2, 3], [3, 3, 2, 1, 3, 1]])
  Set([[2, 2, 1, 2, 3, 1], [3, 1, 2, 3, 3, 2]])
  Set([[1, 3, 3, 3, 1, 2], [3, 1, 2, 2, 3, 2]])
  Set([[2, 1, 3, 1, 3, 3], [2, 3, 2, 3, 2, 1]])
  Set([[1, 2, 1, 1, 3, 2], [2, 3, 1, 2, 2, 3]])
  Set([[3, 1, 2, 3, 2, 2], [2, 1, 3, 2, 1, 1]])
  Set([[2, 2, 3, 1, 2, 3], [1, 3, 2, 1, 1, 2]])
  Set([[1, 3, 2, 2, 1, 2], [3, 1, 1, 2, 3, 1]])
  Set([[3, 2, 2, 3, 1, 3], [3, 1, 1, 2, 2, 2]])
  Set([[2, 1, 2, 1, 2, 3], [3, 2, 1, 2, 3, 3]])
  Set([[3, 3, 1, 1, 2, 3], [1, 3, 2, 2, 1, 1]])
  Set([[2, 1, 1, 2, 3, 1], [3, 2, 2, 3, 2, 1]])
  Set([[2, 1, 1, 3, 3, 1], [3, 3, 3, 2, 2, 1]])
  Set([[1, 1, 2, 3, 3, 3], [2, 2, 1, 1, 3, 1]])
  Set([[1, 1, 1, 3, 3, 2], [2, 3, 2, 1, 1, 2]])
  Set([[2, 1, 1, 1, 3, 2], [1, 3, 3, 2, 3, 1]])
  Set([[1, 2, 3, 3, 1, 3], [2, 1, 3, 1, 2, 1

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[2, 3, 3, 1, 3, 2], [3, 1, 2, 3, 3, 2]])
  Set([[3, 3, 2, 3, 2, 1], [3, 1, 2, 2, 3, 3]])
  Set([[2, 1, 1, 3, 3, 3], [3, 1, 3, 3, 1, 2]])
  Set([[2, 3, 1, 1, 3, 1], [1, 3, 1, 2, 1, 3]])
  Set([[1, 1, 2, 3, 3, 3], [3, 1, 3, 2, 3, 1]])
  Set([[2, 1, 3, 1, 1, 3], [1, 3, 1, 2, 1, 3]])
  Set([[3, 1, 3, 3, 2, 1], [3, 1, 2, 1, 3, 3]])
  Set([[3, 3, 1, 1, 1, 2], [1, 3, 1, 2, 3, 1]])
  Set([[2, 1, 1, 1, 3, 2], [1, 2, 3, 1, 1, 2]])
  Set([[2, 3, 3, 3, 2, 1], [2, 2, 3, 1, 3, 3]])
  Set([[2, 2, 1, 1, 3, 1], [2, 1, 3, 1, 1, 2]])
  Set([[3, 3, 1, 2, 3, 1], [3, 2, 3, 3, 1, 1]])
  Set([[1, 1, 1, 2, 3, 2], [1, 3, 2, 1, 1, 2]])
  Set([[1, 2, 3, 3, 2, 2], [2, 1, 2, 3, 2, 3]])
  Set([[2, 1, 3, 3, 3, 2], [2, 3, 2, 3, 1, 3]])
  Set([[1, 3, 1, 1, 3, 2], [2, 1, 1, 3, 3, 1]])
  Set([[3, 1, 3, 2, 2, 2], [2, 2, 3, 1, 2, 3]])
  Set([[1, 1, 3, 2, 3, 3], [1, 3, 1, 3, 2, 3]])
  Set([[1, 1, 3, 1, 2, 3], [2, 3, 3, 1, 1, 1]])
  Set([[3, 2, 1, 1, 2, 1], [1, 1, 3, 2, 2, 1

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[1, 2, 1, 3, 1, 3], [3, 3, 2, 2, 2, 1]])
  Set([[1, 1, 2, 1, 2, 3], [2, 3, 1, 3, 3, 2]])
  Set([[3, 2, 3, 3, 2, 1], [1, 1, 1, 2, 3, 2]])
  Set([[1, 2, 3, 3, 2, 2], [3, 3, 2, 1, 1, 1]])
  Set([[1, 2, 2, 3, 3, 3], [2, 3, 1, 2, 1, 1]])
  Set([[3, 2, 3, 2, 1, 3], [2, 3, 1, 1, 2, 1]])
  Set([[2, 2, 3, 2, 1, 1], [3, 1, 1, 3, 2, 3]])
  Set([[1, 1, 2, 1, 3, 3], [2, 2, 3, 3, 2, 1]])
  Set([[2, 1, 1, 2, 3, 2], [3, 3, 2, 1, 1, 3]])
  Set([[3, 1, 3, 3, 1, 2], [1, 3, 2, 2, 2, 1]])
  Set([[2, 2, 2, 1, 3, 1], [3, 3, 1, 3, 1, 2]])
  Set([[2, 3, 2, 1, 2, 3], [3, 1, 1, 3, 1, 2]])
  Set([[2, 2, 1, 3, 3, 3], [3, 1, 2, 1, 2, 1]])
  Set([[1, 2, 2, 3, 2, 1], [3, 3, 3, 1, 1, 2]])
  Set([[3, 3, 1, 1, 2, 1], [1, 2, 2, 3, 3, 2]])
  Set([[2, 1, 1, 3, 2, 2], [3, 3, 2, 1, 1, 3]])
  Set([[1, 2, 3, 3, 3, 1], [3, 1, 2, 2, 1, 2]])
  Set([[2, 2, 1, 2, 3, 1], [3, 3, 3, 1, 1, 2]])
  Set([[1, 2, 1, 2, 1, 3], [3, 1, 2, 3, 3, 2]])
  Set([[2, 2, 1, 2, 3, 3], [1, 1, 3, 3, 2, 1

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[3, 1, 1, 1, 3, 2], [3, 2, 1, 3, 3, 1]])
  Set([[3, 2, 3, 1, 3, 1], [3, 3, 2, 1, 1, 1]])
  Set([[1, 1, 1, 2, 3, 3], [1, 3, 2, 1, 3, 3]])
  Set([[3, 2, 1, 3, 1, 3], [2, 3, 1, 3, 1, 1]])
  Set([[2, 3, 1, 2, 1, 2], [2, 2, 1, 3, 1, 1]])
  Set([[1, 1, 2, 3, 2, 2], [1, 1, 1, 2, 2, 3]])
  Set([[1, 1, 3, 2, 2, 2], [1, 1, 2, 1, 3, 2]])
  Set([[3, 3, 1, 1, 2, 3], [1, 3, 1, 2, 1, 3]])
  Set([[3, 2, 1, 2, 1, 2], [1, 2, 1, 2, 3, 1]])
  Set([[2, 3, 1, 3, 3, 1], [3, 3, 1, 2, 1, 1]])
  Set([[3, 1, 1, 3, 2, 1], [3, 1, 3, 3, 1, 2]])
  Set([[1, 1, 3, 2, 1, 3], [1, 2, 3, 1, 3, 3]])
  Set([[2, 2, 1, 2, 3, 1], [2, 1, 1, 3, 2, 1]])
  Set([[1, 3, 3, 1, 1, 2], [1, 3, 2, 3, 1, 3]])
  Set([[2, 2, 1, 3, 1, 1], [2, 2, 3, 1, 2, 1]])
  Set([[3, 2, 2, 1, 2, 1], [1, 1, 2, 1, 2, 3]])
  Set([[3, 2, 1, 1, 1, 2], [1, 2, 1, 3, 2, 2]])
  Set([[2, 3, 1, 1, 1, 3], [1, 3, 3, 2, 1, 3]])
  Set([[1, 2, 3, 3, 2, 3], [2, 1, 3, 2, 2, 3]])
  Set([[1, 2, 3, 3, 1, 3], [1, 1, 3, 3, 2, 1

Set{Set{Vector{Int64}}} with 4320 elements:
  Set([[2, 1, 1, 3, 3, 1], [2, 2, 1, 2, 1, 3]])
  Set([[1, 2, 3, 3, 1, 3], [3, 2, 2, 1, 2, 3]])
  Set([[2, 1, 1, 1, 3, 3], [3, 1, 2, 3, 3, 2]])
  Set([[1, 2, 2, 2, 3, 1], [3, 1, 3, 2, 1, 1]])
  Set([[1, 1, 2, 3, 3, 3], [2, 1, 1, 2, 1, 3]])
  Set([[1, 1, 3, 2, 2, 2], [3, 1, 1, 3, 2, 1]])
  Set([[3, 3, 2, 1, 1, 3], [1, 2, 1, 1, 2, 3]])
  Set([[2, 3, 1, 2, 3, 3], [3, 3, 1, 1, 2, 1]])
  Set([[1, 2, 1, 1, 3, 2], [1, 2, 3, 2, 2, 3]])
  Set([[3, 1, 3, 1, 2, 1], [1, 3, 2, 2, 2, 1]])
  Set([[2, 1, 3, 1, 2, 1], [2, 2, 2, 1, 3, 3]])
  Set([[1, 1, 1, 3, 3, 2], [1, 3, 2, 1, 2, 2]])
  Set([[1, 1, 2, 3, 3, 1], [3, 1, 3, 2, 3, 2]])
  Set([[3, 1, 1, 1, 2, 2], [3, 2, 3, 1, 1, 3]])
  Set([[2, 3, 3, 2, 3, 1], [1, 3, 1, 3, 2, 1]])
  Set([[3, 3, 3, 2, 1, 2], [1, 3, 2, 1, 2, 2]])
  Set([[2, 2, 3, 3, 1, 3], [1, 2, 1, 2, 2, 3]])
  Set([[2, 1, 1, 1, 3, 2], [2, 1, 3, 2, 2, 3]])
  Set([[2, 1, 3, 2, 1, 1], [3, 3, 3, 1, 1, 2]])
  Set([[3, 2, 2, 2, 1, 3], [3, 2, 1, 3, 3, 1

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[1, 2, 1, 2, 2, 3], [1, 3, 2, 2, 1, 1]])
  Set([[3, 3, 2, 2, 1, 3], [1, 3, 2, 3, 2, 2]])
  Set([[2, 2, 1, 1, 3, 1], [2, 1, 3, 2, 2, 1]])
  Set([[1, 2, 1, 3, 1, 3], [2, 3, 3, 3, 1, 1]])
  Set([[3, 2, 2, 1, 2, 1], [1, 1, 2, 1, 3, 2]])
  Set([[2, 2, 2, 3, 3, 1], [2, 3, 1, 3, 2, 3]])
  Set([[1, 3, 3, 2, 2, 3], [2, 1, 2, 2, 3, 3]])
  Set([[2, 3, 3, 1, 3, 1], [1, 3, 1, 3, 2, 1]])
  Set([[1, 2, 1, 3, 2, 2], [1, 3, 2, 1, 1, 2]])
  Set([[1, 3, 2, 1, 1, 3], [1, 1, 3, 3, 2, 3]])
  Set([[2, 3, 1, 1, 3, 1], [3, 3, 3, 1, 1, 2]])
  Set([[3, 1, 3, 1, 3, 2], [2, 3, 1, 1, 3, 1]])
  Set([[1, 3, 3, 1, 2, 1], [1, 3, 1, 3, 3, 2]])
  Set([[3, 1, 2, 3, 1, 3], [2, 3, 1, 1, 1, 3]])
  Set([[2, 1, 3, 2, 1, 1], [2, 3, 2, 1, 1, 2]])
  Set([[2, 3, 3, 2, 2, 1], [3, 3, 2, 1, 2, 3]])
  Set([[3, 1, 1, 2, 3, 3], [2, 3, 1, 1, 1, 3]])
  Set([[2, 1, 3, 1, 3, 1], [3, 2, 3, 1, 1, 3]])
  Set([[2, 1, 1, 1, 3, 2], [2, 1, 2, 3, 2, 1]])
  Set([[1, 3, 2, 3, 3, 1], [3, 3, 1, 2, 1, 1

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[2, 3, 1, 2, 1, 2], [2, 3, 3, 1, 3, 2]])
  Set([[3, 1, 3, 3, 1, 2], [1, 2, 3, 3, 2, 2]])
  Set([[1, 1, 3, 2, 3, 3], [2, 2, 3, 2, 1, 3]])
  Set([[2, 1, 1, 3, 3, 3], [1, 1, 1, 2, 2, 3]])
  Set([[1, 3, 1, 2, 2, 2], [1, 1, 1, 3, 3, 2]])
  Set([[2, 3, 3, 3, 2, 1], [1, 3, 3, 2, 1, 1]])
  Set([[3, 3, 1, 3, 2, 2], [1, 3, 2, 1, 2, 2]])
  Set([[2, 3, 3, 2, 1, 3], [2, 1, 1, 2, 2, 3]])
  Set([[3, 2, 1, 2, 1, 2], [1, 2, 1, 3, 1, 3]])
  Set([[2, 1, 3, 3, 3, 2], [1, 1, 3, 2, 3, 1]])
  Set([[2, 1, 2, 1, 3, 2], [2, 3, 1, 3, 3, 2]])
  Set([[1, 3, 3, 2, 2, 3], [2, 1, 3, 2, 2, 1]])
  Set([[1, 1, 3, 3, 2, 3], [2, 2, 3, 3, 2, 1]])
  Set([[3, 2, 1, 2, 1, 2], [1, 3, 1, 3, 1, 2]])
  Set([[2, 2, 1, 2, 1, 3], [3, 2, 1, 3, 1, 1]])
  Set([[3, 1, 3, 2, 3, 1], [1, 2, 3, 2, 3, 2]])
  Set([[2, 1, 3, 2, 1, 1], [2, 1, 2, 2, 3, 3]])
  Set([[2, 3, 2, 1, 2, 3], [1, 3, 1, 3, 2, 3]])
  Set([[1, 2, 3, 2, 1, 1], [1, 3, 3, 3, 2, 1]])
  Set([[3, 3, 1, 1, 2, 3], [2, 3, 1, 1, 1, 2

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[1, 3, 3, 2, 1, 1], [2, 3, 2, 2, 1, 1]])
  Set([[2, 1, 2, 2, 3, 1], [1, 1, 3, 2, 3, 1]])
  Set([[1, 2, 3, 1, 2, 2], [3, 2, 3, 1, 2, 3]])
  Set([[3, 3, 1, 3, 2, 1], [2, 3, 1, 1, 2, 1]])
  Set([[2, 2, 2, 1, 3, 1], [2, 2, 3, 1, 3, 3]])
  Set([[3, 2, 3, 2, 3, 1], [3, 2, 2, 2, 1, 1]])
  Set([[1, 1, 1, 2, 3, 3], [3, 2, 1, 2, 3, 3]])
  Set([[3, 3, 2, 3, 2, 1], [3, 1, 2, 2, 2, 1]])
  Set([[1, 2, 3, 1, 3, 3], [2, 2, 2, 1, 3, 3]])
  Set([[1, 2, 3, 2, 1, 2], [1, 1, 3, 2, 1, 3]])
  Set([[1, 2, 3, 2, 3, 3], [1, 2, 1, 2, 2, 3]])
  Set([[1, 1, 2, 3, 1, 2], [2, 1, 2, 3, 3, 2]])
  Set([[2, 3, 3, 1, 2, 3], [2, 3, 2, 1, 2, 1]])
  Set([[2, 3, 3, 1, 1, 1], [2, 3, 3, 3, 2, 1]])
  Set([[3, 1, 2, 1, 1, 3], [2, 1, 2, 2, 1, 3]])
  Set([[2, 3, 1, 2, 3, 2], [2, 3, 1, 1, 3, 3]])
  Set([[2, 2, 1, 3, 1, 2], [2, 1, 1, 3, 1, 3]])
  Set([[3, 2, 1, 2, 2, 1], [3, 3, 1, 1, 2, 1]])
  Set([[2, 1, 1, 1, 3, 2], [2, 3, 2, 1, 3, 2]])
  Set([[2, 3, 2, 1, 2, 3], [3, 3, 1, 1, 2, 3

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[3, 1, 3, 3, 1, 2], [3, 3, 1, 3, 1, 2]])
  Set([[3, 2, 1, 2, 1, 2], [3, 1, 1, 2, 2, 2]])
  Set([[1, 3, 3, 2, 1, 1], [1, 1, 3, 2, 3, 1]])
  Set([[3, 3, 1, 1, 2, 1], [1, 3, 1, 3, 2, 1]])
  Set([[2, 2, 1, 3, 2, 3], [3, 2, 1, 2, 2, 3]])
  Set([[2, 2, 3, 3, 1, 3], [2, 3, 3, 2, 1, 3]])
  Set([[2, 3, 3, 2, 2, 1], [2, 3, 2, 2, 3, 1]])
  Set([[2, 2, 3, 3, 3, 1], [2, 3, 3, 2, 3, 1]])
  Set([[1, 3, 1, 2, 2, 1], [2, 3, 1, 2, 1, 1]])
  Set([[2, 1, 1, 3, 3, 3], [2, 3, 1, 3, 1, 3]])
  Set([[3, 1, 1, 3, 2, 3], [3, 1, 3, 1, 2, 3]])
  Set([[3, 1, 3, 1, 2, 3], [1, 1, 3, 3, 2, 3]])
  Set([[2, 2, 1, 3, 2, 1], [1, 2, 1, 3, 2, 2]])
  Set([[1, 3, 2, 1, 3, 1], [3, 1, 2, 1, 3, 1]])
  Set([[2, 3, 3, 1, 1, 3], [2, 3, 3, 3, 1, 1]])
  Set([[2, 3, 1, 1, 3, 3], [2, 3, 1, 3, 1, 3]])
  Set([[3, 2, 2, 1, 2, 3], [2, 2, 2, 1, 3, 3]])
  Set([[3, 1, 2, 2, 2, 3], [2, 1, 2, 2, 3, 3]])
  Set([[3, 2, 1, 1, 3, 1], [1, 2, 1, 3, 3, 1]])
  Set([[1, 1, 2, 2, 1, 3], [1, 2, 2, 1, 1, 3

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[1, 3, 2, 1, 2, 1], [2, 3, 1, 1, 1, 2]])
  Set([[3, 1, 1, 3, 2, 3], [1, 3, 3, 3, 2, 1]])
  Set([[3, 1, 3, 2, 1, 3], [1, 3, 3, 2, 3, 1]])
  Set([[2, 1, 1, 2, 3, 2], [1, 2, 2, 1, 3, 2]])
  Set([[1, 2, 3, 3, 2, 2], [1, 3, 2, 2, 3, 2]])
  Set([[3, 1, 1, 1, 2, 2], [3, 2, 2, 1, 1, 1]])
  Set([[3, 2, 1, 3, 1, 3], [1, 2, 3, 1, 3, 3]])
  Set([[2, 1, 3, 3, 2, 2], [2, 1, 2, 2, 3, 3]])
  Set([[3, 3, 1, 1, 2, 3], [1, 3, 3, 3, 2, 1]])
  Set([[3, 1, 3, 3, 1, 2], [1, 3, 3, 1, 3, 2]])
  Set([[3, 3, 2, 1, 3, 2], [3, 2, 3, 1, 2, 3]])
  Set([[1, 1, 3, 3, 1, 2], [3, 3, 1, 1, 1, 2]])
  Set([[3, 2, 3, 1, 3, 1], [1, 2, 3, 3, 1, 3]])
  Set([[2, 2, 2, 1, 3, 1], [2, 1, 1, 2, 3, 2]])
  Set([[3, 1, 2, 1, 2, 2], [3, 2, 2, 2, 1, 1]])
  Set([[1, 1, 3, 1, 2, 3], [1, 3, 1, 3, 2, 1]])
  Set([[1, 3, 1, 1, 2, 2], [2, 3, 1, 2, 1, 1]])
  Set([[2, 2, 1, 3, 2, 3], [2, 3, 1, 2, 3, 2]])
  Set([[1, 1, 2, 1, 2, 3], [2, 1, 1, 2, 1, 3]])
  Set([[2, 1, 2, 3, 2, 1], [1, 2, 2, 3, 1, 2]

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[2, 2, 1, 3, 1, 1], [1, 2, 3, 3, 2, 2]])
  Set([[1, 2, 3, 2, 1, 1], [3, 2, 1, 3, 3, 1]])
  Set([[3, 1, 1, 2, 3, 1], [3, 2, 1, 1, 2, 2]])
  Set([[1, 1, 1, 2, 3, 3], [2, 2, 1, 1, 2, 3]])
  Set([[3, 2, 2, 1, 2, 1], [2, 3, 2, 3, 3, 1]])
  Set([[1, 1, 3, 3, 2, 1], [3, 2, 3, 1, 2, 3]])
  Set([[3, 2, 2, 3, 2, 1], [3, 3, 3, 2, 1, 1]])
  Set([[1, 2, 2, 3, 3, 3], [1, 3, 2, 2, 2, 1]])
  Set([[1, 3, 3, 3, 2, 1], [1, 2, 1, 1, 2, 3]])
  Set([[2, 2, 3, 3, 3, 1], [1, 2, 1, 3, 1, 3]])
  Set([[2, 1, 2, 3, 2, 1], [3, 3, 3, 2, 2, 1]])
  Set([[2, 1, 1, 1, 3, 2], [1, 3, 2, 2, 3, 2]])
  Set([[2, 3, 1, 2, 1, 2], [1, 3, 1, 3, 2, 1]])
  Set([[1, 2, 3, 2, 1, 2], [2, 1, 3, 3, 1, 1]])
  Set([[2, 3, 2, 1, 2, 1], [2, 2, 3, 1, 3, 3]])
  Set([[1, 3, 2, 2, 1, 2], [1, 3, 1, 3, 2, 1]])
  Set([[3, 2, 1, 2, 1, 2], [2, 3, 3, 2, 1, 3]])
  Set([[1, 2, 3, 2, 3, 3], [3, 1, 3, 2, 1, 1]])
  Set([[1, 2, 2, 1, 3, 2], [2, 1, 1, 1, 3, 3]])
  Set([[2, 2, 1, 3, 3, 3], [2, 3, 1, 1, 2, 2

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[2, 1, 2, 3, 1, 2], [3, 3, 2, 1, 3, 1]])
  Set([[1, 2, 1, 2, 3, 2], [3, 2, 3, 1, 1, 3]])
  Set([[3, 1, 2, 3, 2, 3], [3, 2, 1, 2, 1, 1]])
  Set([[2, 1, 3, 1, 2, 1], [1, 3, 3, 2, 3, 2]])
  Set([[1, 1, 2, 1, 3, 2], [2, 2, 1, 3, 3, 3]])
  Set([[1, 2, 1, 2, 2, 3], [3, 3, 2, 1, 1, 3]])
  Set([[2, 3, 3, 1, 2, 2], [2, 1, 1, 3, 1, 3]])
  Set([[1, 2, 1, 1, 3, 3], [3, 3, 1, 2, 2, 2]])
  Set([[1, 1, 2, 1, 2, 3], [3, 1, 3, 2, 3, 2]])
  Set([[2, 1, 3, 3, 3, 2], [3, 1, 2, 1, 2, 1]])
  Set([[3, 2, 1, 2, 1, 1], [2, 3, 1, 3, 2, 3]])
  Set([[2, 2, 3, 1, 2, 1], [1, 2, 1, 3, 3, 3]])
  Set([[2, 3, 1, 2, 3, 3], [1, 1, 2, 1, 3, 2]])
  Set([[3, 1, 2, 3, 1, 1], [2, 2, 2, 1, 3, 3]])
  Set([[1, 3, 3, 1, 2, 1], [3, 2, 1, 3, 2, 2]])
  Set([[3, 2, 2, 2, 1, 3], [1, 3, 1, 3, 1, 2]])
  Set([[3, 2, 3, 1, 3, 1], [1, 1, 2, 2, 3, 2]])
  Set([[1, 2, 3, 2, 1, 2], [3, 1, 3, 3, 2, 1]])
  Set([[2, 1, 2, 3, 2, 3], [1, 1, 3, 2, 3, 1]])
  Set([[1, 1, 3, 2, 2, 2], [3, 3, 1, 1, 3, 2

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[1, 1, 1, 2, 3, 2], [1, 1, 3, 3, 2, 3]])
  Set([[2, 1, 2, 1, 1, 3], [2, 3, 2, 3, 2, 1]])
  Set([[1, 3, 1, 1, 3, 2], [1, 2, 1, 2, 2, 3]])
  Set([[2, 2, 1, 2, 3, 3], [2, 1, 3, 2, 1, 1]])
  Set([[1, 3, 1, 2, 1, 2], [3, 1, 2, 2, 3, 2]])
  Set([[3, 3, 1, 1, 1, 2], [2, 2, 1, 2, 1, 3]])
  Set([[2, 2, 1, 3, 2, 1], [2, 2, 3, 1, 3, 3]])
  Set([[2, 2, 1, 3, 1, 2], [2, 3, 3, 1, 3, 2]])
  Set([[2, 1, 2, 1, 1, 3], [2, 3, 2, 2, 3, 1]])
  Set([[3, 1, 1, 2, 1, 2], [1, 2, 3, 2, 3, 2]])
  Set([[3, 1, 2, 2, 1, 2], [1, 3, 2, 3, 3, 2]])
  Set([[1, 3, 3, 1, 2, 1], [3, 3, 3, 2, 1, 2]])
  Set([[2, 1, 2, 2, 3, 1], [3, 1, 3, 1, 2, 1]])
  Set([[2, 2, 3, 1, 2, 1], [2, 3, 1, 3, 2, 3]])
  Set([[2, 1, 2, 3, 1, 2], [2, 3, 3, 1, 3, 2]])
  Set([[1, 3, 3, 1, 1, 2], [1, 2, 2, 1, 2, 3]])
  Set([[3, 1, 3, 3, 2, 2], [3, 2, 3, 1, 1, 1]])
  Set([[3, 1, 1, 2, 3, 1], [3, 2, 3, 1, 3, 2]])
  Set([[3, 1, 2, 1, 1, 3], [3, 2, 1, 2, 3, 3]])
  Set([[2, 1, 2, 3, 3, 3], [2, 3, 2, 1, 2, 1

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[1, 2, 1, 3, 1, 3], [3, 1, 2, 3, 2, 2]])
  Set([[3, 3, 1, 3, 1, 2], [2, 1, 2, 2, 1, 3]])
  Set([[2, 1, 3, 2, 2, 1], [3, 3, 2, 1, 3, 1]])
  Set([[2, 1, 1, 2, 3, 2], [1, 1, 3, 3, 2, 3]])
  Set([[3, 3, 1, 3, 1, 2], [2, 2, 2, 1, 1, 3]])
  Set([[2, 2, 3, 2, 3, 1], [3, 1, 3, 1, 1, 2]])
  Set([[1, 3, 3, 1, 2, 1], [2, 2, 3, 2, 1, 3]])
  Set([[1, 1, 3, 2, 2, 2], [3, 1, 2, 3, 1, 3]])
  Set([[3, 1, 1, 1, 2, 2], [1, 3, 3, 2, 2, 3]])
  Set([[1, 2, 2, 3, 2, 3], [2, 1, 3, 3, 1, 1]])
  Set([[2, 2, 1, 2, 3, 3], [3, 1, 2, 1, 1, 3]])
  Set([[2, 3, 3, 2, 2, 1], [1, 1, 3, 1, 3, 2]])
  Set([[2, 1, 3, 2, 1, 2], [1, 3, 2, 3, 1, 3]])
  Set([[1, 3, 2, 1, 2, 1], [3, 1, 3, 2, 2, 3]])
  Set([[3, 2, 2, 3, 1, 2], [1, 1, 1, 3, 2, 3]])
  Set([[1, 3, 1, 1, 3, 2], [2, 2, 3, 2, 3, 1]])
  Set([[1, 1, 2, 2, 3, 1], [2, 3, 2, 3, 1, 3]])
  Set([[2, 3, 1, 2, 2, 3], [1, 1, 2, 3, 1, 3]])
  Set([[1, 3, 2, 2, 1, 2], [1, 2, 3, 3, 3, 1]])
  Set([[2, 3, 3, 1, 1, 3], [3, 2, 1, 1, 2, 2

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[3, 3, 1, 2, 1, 3], [1, 3, 3, 1, 2, 1]])
  Set([[1, 1, 3, 3, 3, 2], [2, 3, 1, 3, 1, 1]])
  Set([[1, 1, 3, 2, 1, 2], [3, 2, 1, 1, 2, 2]])
  Set([[3, 3, 2, 1, 2, 3], [3, 2, 3, 2, 1, 2]])
  Set([[3, 1, 2, 3, 2, 3], [2, 3, 3, 1, 2, 2]])
  Set([[1, 3, 3, 1, 1, 2], [2, 1, 3, 3, 3, 1]])
  Set([[1, 1, 2, 1, 3, 3], [3, 3, 3, 1, 1, 2]])
  Set([[2, 2, 1, 2, 3, 1], [2, 1, 3, 1, 1, 2]])
  Set([[3, 1, 1, 1, 2, 2], [1, 3, 2, 2, 2, 1]])
  Set([[3, 1, 3, 1, 3, 2], [1, 1, 1, 3, 2, 3]])
  Set([[1, 2, 3, 3, 1, 3], [3, 3, 1, 1, 1, 2]])
  Set([[3, 3, 2, 2, 1, 3], [2, 1, 2, 3, 3, 2]])
  Set([[3, 1, 1, 2, 3, 3], [2, 3, 1, 3, 1, 1]])
  Set([[1, 3, 3, 2, 3, 2], [2, 3, 2, 3, 2, 1]])
  Set([[2, 2, 3, 2, 1, 1], [1, 1, 1, 2, 3, 2]])
  Set([[3, 3, 1, 3, 2, 1], [1, 3, 2, 1, 1, 3]])
  Set([[2, 3, 2, 1, 2, 3], [3, 2, 3, 2, 1, 3]])
  Set([[3, 2, 1, 2, 2, 1], [2, 1, 2, 3, 1, 1]])
  Set([[1, 2, 3, 1, 2, 2], [2, 2, 1, 3, 1, 1]])
  Set([[3, 3, 1, 3, 2, 1], [2, 1, 1, 1, 3, 3

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[1, 1, 1, 2, 3, 3], [3, 1, 2, 2, 3, 2]])
  Set([[3, 1, 3, 1, 2, 3], [2, 2, 3, 1, 2, 1]])
  Set([[2, 1, 1, 3, 3, 3], [2, 2, 1, 1, 2, 3]])
  Set([[2, 2, 3, 3, 1, 2], [3, 2, 1, 3, 1, 1]])
  Set([[2, 1, 1, 3, 3, 3], [2, 1, 2, 1, 2, 3]])
  Set([[2, 1, 3, 2, 2, 1], [1, 3, 3, 2, 3, 1]])
  Set([[3, 1, 3, 2, 2, 2], [1, 1, 3, 2, 3, 1]])
  Set([[1, 2, 3, 3, 2, 3], [1, 1, 2, 3, 2, 1]])
  Set([[1, 1, 3, 2, 2, 2], [3, 1, 3, 2, 1, 3]])
  Set([[1, 3, 2, 3, 2, 3], [1, 3, 1, 1, 2, 2]])
  Set([[1, 2, 3, 3, 2, 2], [1, 3, 1, 3, 2, 1]])
  Set([[3, 1, 2, 1, 1, 2], [3, 3, 3, 2, 1, 2]])
  Set([[3, 2, 1, 1, 3, 3], [2, 2, 1, 2, 3, 1]])
  Set([[2, 1, 1, 3, 2, 2], [1, 3, 1, 3, 2, 3]])
  Set([[3, 1, 2, 3, 2, 2], [3, 1, 1, 1, 3, 2]])
  Set([[1, 3, 1, 2, 3, 1], [3, 3, 2, 2, 2, 1]])
  Set([[1, 1, 2, 3, 1, 3], [3, 2, 2, 3, 1, 2]])
  Set([[1, 1, 2, 1, 3, 2], [3, 1, 3, 2, 3, 2]])
  Set([[1, 2, 2, 1, 3, 1], [1, 3, 2, 3, 3, 2]])
  Set([[2, 3, 3, 3, 1, 1], [2, 3, 1, 2, 2, 1

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[2, 1, 1, 1, 3, 2], [1, 2, 3, 3, 1, 1]])
  Set([[3, 2, 1, 1, 2, 1], [1, 1, 3, 2, 1, 3]])
  Set([[2, 1, 2, 3, 3, 3], [3, 3, 3, 2, 1, 1]])
  Set([[2, 1, 3, 3, 1, 1], [1, 3, 1, 1, 2, 2]])
  Set([[3, 1, 3, 1, 3, 2], [2, 3, 1, 3, 2, 3]])
  Set([[3, 1, 2, 3, 1, 1], [1, 3, 1, 1, 2, 2]])
  Set([[2, 3, 2, 3, 1, 2], [3, 2, 1, 2, 2, 1]])
  Set([[1, 1, 2, 1, 3, 3], [2, 3, 1, 2, 1, 1]])
  Set([[1, 1, 3, 3, 2, 1], [2, 3, 1, 1, 1, 2]])
  Set([[1, 2, 1, 2, 3, 1], [3, 1, 3, 1, 1, 2]])
  Set([[3, 2, 2, 2, 1, 3], [2, 3, 1, 1, 2, 2]])
  Set([[3, 1, 2, 1, 2, 2], [2, 2, 3, 2, 1, 3]])
  Set([[3, 1, 3, 2, 3, 2], [2, 3, 1, 3, 1, 3]])
  Set([[2, 3, 2, 3, 3, 1], [3, 1, 3, 1, 2, 3]])
  Set([[1, 1, 3, 1, 2, 3], [2, 2, 1, 3, 1, 1]])
  Set([[2, 1, 3, 3, 3, 2], [3, 3, 1, 1, 2, 3]])
  Set([[2, 3, 1, 2, 3, 3], [3, 1, 3, 3, 2, 1]])
  Set([[2, 3, 3, 2, 1, 3], [3, 2, 1, 3, 3, 1]])
  Set([[1, 2, 1, 3, 1, 2], [3, 1, 2, 1, 3, 1]])
  Set([[1, 3, 1, 2, 1, 2], [2, 1, 3, 1, 3, 1]

Set{Set{Vector{Int64}}} with 360 elements:
  Set([[3, 3, 3, 2, 2, 1], [3, 3, 3, 2, 1, 1]])
  Set([[2, 1, 3, 1, 3, 3], [2, 1, 3, 2, 3, 3]])
  Set([[1, 2, 2, 3, 1, 2], [3, 2, 2, 3, 1, 2]])
  Set([[3, 3, 1, 2, 2, 2], [3, 1, 1, 2, 2, 2]])
  Set([[3, 2, 3, 1, 2, 3], [3, 2, 3, 1, 1, 3]])
  Set([[1, 3, 2, 3, 3, 2], [1, 3, 1, 3, 3, 2]])
  Set([[2, 2, 3, 1, 3, 2], [2, 2, 1, 1, 3, 2]])
  Set([[1, 1, 2, 2, 3, 1], [1, 1, 2, 3, 3, 1]])
  Set([[1, 2, 3, 3, 2, 2], [1, 2, 1, 3, 2, 2]])
  Set([[1, 1, 3, 1, 2, 3], [1, 1, 2, 1, 2, 3]])
  Set([[2, 1, 3, 2, 1, 2], [2, 1, 3, 2, 3, 2]])
  Set([[1, 1, 1, 2, 3, 2], [1, 1, 1, 3, 3, 2]])
  Set([[2, 3, 3, 2, 3, 1], [1, 3, 3, 2, 3, 1]])
  Set([[1, 2, 2, 1, 2, 3], [3, 2, 2, 1, 2, 3]])
  Set([[2, 3, 2, 3, 1, 2], [2, 3, 2, 1, 1, 2]])
  Set([[1, 3, 2, 1, 1, 3], [1, 2, 2, 1, 1, 3]])
  Set([[1, 2, 3, 1, 2, 1], [1, 2, 3, 1, 3, 1]])
  Set([[1, 3, 2, 3, 2, 3], [1, 3, 1, 3, 2, 3]])
  Set([[1, 3, 1, 2, 1, 2], [1, 3, 1, 3, 1, 2]])
  Set([[2, 3, 3, 2, 2, 1], [2, 1, 3, 2, 2, 1]

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[1, 3, 2, 3, 2, 2], [3, 2, 3, 3, 1, 1]])
  Set([[3, 2, 1, 2, 1, 2], [2, 1, 3, 3, 3, 2]])
  Set([[2, 2, 1, 2, 3, 3], [1, 3, 2, 2, 1, 1]])
  Set([[2, 1, 3, 1, 1, 2], [3, 3, 1, 1, 2, 3]])
  Set([[2, 1, 2, 1, 1, 3], [2, 2, 1, 3, 3, 2]])
  Set([[1, 1, 2, 1, 3, 2], [3, 1, 3, 2, 1, 3]])
  Set([[3, 1, 2, 1, 1, 2], [2, 3, 2, 2, 3, 1]])
  Set([[1, 1, 3, 1, 2, 2], [3, 2, 2, 3, 1, 2]])
  Set([[2, 2, 3, 2, 1, 1], [3, 1, 1, 3, 1, 2]])
  Set([[1, 1, 2, 3, 3, 3], [2, 2, 3, 2, 3, 1]])
  Set([[1, 3, 1, 3, 1, 2], [2, 2, 1, 2, 3, 1]])
  Set([[2, 1, 3, 3, 3, 2], [2, 2, 2, 1, 1, 3]])
  Set([[3, 1, 2, 3, 1, 1], [1, 2, 3, 3, 3, 2]])
  Set([[2, 1, 1, 3, 3, 3], [3, 2, 2, 2, 3, 1]])
  Set([[3, 3, 1, 3, 2, 1], [2, 2, 3, 1, 1, 1]])
  Set([[1, 3, 1, 2, 3, 1], [2, 3, 2, 3, 1, 3]])
  Set([[1, 1, 3, 3, 2, 3], [2, 2, 2, 1, 3, 3]])
  Set([[1, 3, 1, 2, 3, 1], [1, 2, 2, 1, 2, 3]])
  Set([[3, 3, 1, 3, 2, 1], [1, 2, 1, 2, 1, 3]])
  Set([[2, 1, 2, 2, 3, 1], [1, 3, 3, 2, 2, 3

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[2, 1, 2, 1, 3, 2], [2, 2, 2, 1, 3, 3]])
  Set([[2, 3, 2, 3, 2, 1], [2, 3, 2, 2, 1, 1]])
  Set([[2, 2, 1, 1, 3, 1], [1, 2, 3, 1, 3, 1]])
  Set([[1, 1, 3, 3, 2, 3], [1, 3, 2, 3, 2, 3]])
  Set([[2, 2, 2, 3, 1, 1], [2, 3, 2, 3, 2, 1]])
  Set([[3, 1, 1, 1, 2, 2], [3, 1, 1, 3, 2, 1]])
  Set([[3, 2, 2, 3, 3, 1], [3, 3, 2, 1, 3, 1]])
  Set([[3, 3, 2, 1, 1, 1], [1, 3, 2, 2, 1, 1]])
  Set([[3, 3, 1, 1, 3, 2], [3, 3, 3, 1, 2, 2]])
  Set([[2, 2, 2, 3, 1, 1], [3, 2, 2, 3, 2, 1]])
  Set([[2, 2, 1, 3, 2, 1], [2, 3, 2, 3, 2, 1]])
  Set([[3, 3, 1, 3, 2, 1], [3, 3, 3, 2, 2, 1]])
  Set([[3, 1, 2, 3, 1, 1], [1, 2, 2, 3, 1, 1]])
  Set([[3, 2, 3, 1, 3, 1], [2, 2, 3, 3, 3, 1]])
  Set([[1, 3, 1, 2, 3, 1], [2, 3, 1, 2, 1, 1]])
  Set([[2, 3, 1, 2, 3, 2], [2, 3, 1, 1, 2, 2]])
  Set([[3, 3, 2, 3, 2, 1], [3, 3, 3, 1, 2, 1]])
  Set([[2, 1, 3, 2, 3, 2], [2, 1, 3, 2, 2, 1]])
  Set([[1, 2, 1, 3, 2, 1], [3, 2, 1, 3, 1, 1]])
  Set([[3, 1, 1, 2, 1, 3], [1, 1, 1, 2, 2, 3

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[3, 2, 3, 2, 3, 1], [3, 2, 1, 2, 3, 3]])
  Set([[1, 2, 2, 3, 2, 1], [1, 3, 2, 2, 2, 1]])
  Set([[2, 2, 1, 2, 3, 3], [2, 2, 2, 1, 3, 3]])
  Set([[2, 3, 1, 3, 3, 1], [3, 3, 1, 3, 2, 1]])
  Set([[2, 2, 3, 3, 2, 1], [2, 1, 3, 3, 2, 2]])
  Set([[2, 2, 3, 1, 1, 2], [3, 2, 2, 1, 1, 2]])
  Set([[1, 3, 2, 3, 3, 2], [3, 3, 2, 3, 1, 2]])
  Set([[1, 2, 1, 3, 1, 3], [1, 1, 1, 3, 2, 3]])
  Set([[1, 2, 3, 1, 3, 3], [1, 3, 3, 1, 3, 2]])
  Set([[3, 1, 2, 3, 2, 2], [3, 2, 1, 3, 2, 2]])
  Set([[3, 2, 3, 1, 1, 3], [3, 3, 3, 1, 1, 2]])
  Set([[1, 2, 2, 3, 3, 3], [3, 2, 2, 3, 3, 1]])
  Set([[2, 1, 2, 3, 3, 2], [2, 2, 2, 3, 3, 1]])
  Set([[3, 1, 2, 1, 1, 3], [3, 1, 1, 1, 2, 3]])
  Set([[3, 3, 1, 2, 2, 2], [3, 3, 2, 2, 2, 1]])
  Set([[1, 3, 1, 1, 3, 2], [1, 3, 2, 1, 3, 1]])
  Set([[1, 3, 3, 2, 1, 1], [1, 3, 3, 1, 1, 2]])
  Set([[1, 2, 3, 3, 1, 1], [1, 1, 3, 3, 1, 2]])
  Set([[3, 2, 3, 2, 2, 1], [3, 2, 3, 2, 1, 2]])
  Set([[1, 1, 1, 3, 2, 2], [1, 3, 1, 1, 2, 2]

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[3, 2, 2, 3, 3, 1], [2, 1, 1, 2, 1, 3]])
  Set([[2, 2, 1, 3, 3, 3], [1, 1, 3, 2, 2, 1]])
  Set([[1, 1, 2, 2, 3, 1], [3, 2, 3, 3, 1, 2]])
  Set([[1, 1, 1, 3, 2, 2], [2, 3, 2, 1, 3, 3]])
  Set([[3, 1, 1, 3, 1, 2], [2, 3, 2, 2, 3, 1]])
  Set([[3, 2, 1, 3, 1, 3], [1, 3, 2, 1, 2, 2]])
  Set([[2, 2, 2, 3, 1, 1], [3, 1, 1, 2, 3, 3]])
  Set([[2, 1, 3, 1, 1, 3], [1, 2, 2, 3, 3, 2]])
  Set([[2, 2, 1, 1, 3, 1], [3, 3, 3, 2, 1, 2]])
  Set([[3, 2, 1, 2, 1, 2], [2, 3, 3, 1, 3, 1]])
  Set([[1, 3, 1, 1, 3, 2], [2, 2, 3, 3, 2, 1]])
  Set([[1, 3, 3, 3, 1, 2], [2, 2, 1, 1, 2, 3]])
  Set([[3, 3, 2, 1, 2, 3], [1, 2, 1, 3, 1, 2]])
  Set([[1, 2, 1, 3, 3, 1], [3, 1, 2, 2, 2, 3]])
  Set([[1, 3, 2, 3, 2, 2], [2, 1, 1, 1, 3, 3]])
  Set([[2, 1, 1, 1, 2, 3], [3, 2, 2, 3, 3, 1]])
  Set([[3, 3, 1, 2, 2, 3], [2, 2, 3, 1, 1, 1]])
  Set([[1, 2, 2, 3, 1, 1], [3, 3, 3, 1, 2, 2]])
  Set([[3, 1, 3, 2, 2, 2], [1, 2, 1, 3, 1, 3]])
  Set([[2, 2, 2, 1, 3, 1], [1, 3, 1, 3, 2, 3]

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[2, 1, 3, 1, 2, 1], [2, 2, 3, 1, 2, 1]])
  Set([[2, 3, 2, 1, 1, 2], [1, 3, 2, 1, 1, 2]])
  Set([[1, 2, 1, 2, 3, 2], [1, 2, 1, 2, 3, 1]])
  Set([[2, 1, 1, 3, 2, 2], [2, 1, 1, 3, 2, 1]])
  Set([[1, 3, 3, 3, 2, 1], [1, 3, 1, 3, 2, 1]])
  Set([[3, 3, 1, 2, 1, 1], [3, 3, 3, 2, 1, 1]])
  Set([[2, 2, 2, 3, 1, 3], [3, 2, 2, 3, 1, 3]])
  Set([[2, 3, 3, 3, 1, 2], [2, 3, 2, 3, 1, 2]])
  Set([[2, 3, 1, 2, 3, 2], [2, 3, 1, 3, 3, 2]])
  Set([[3, 2, 3, 2, 3, 1], [3, 2, 2, 2, 3, 1]])
  Set([[3, 2, 2, 2, 3, 1], [3, 2, 2, 3, 3, 1]])
  Set([[3, 1, 2, 3, 2, 3], [3, 1, 2, 3, 2, 2]])
  Set([[1, 3, 3, 2, 3, 2], [1, 2, 3, 2, 3, 2]])
  Set([[1, 3, 1, 2, 3, 3], [1, 3, 1, 2, 3, 1]])
  Set([[2, 1, 2, 3, 1, 2], [1, 1, 2, 3, 1, 2]])
  Set([[3, 1, 3, 2, 3, 2], [3, 1, 2, 2, 3, 2]])
  Set([[1, 2, 3, 1, 1, 2], [1, 2, 3, 2, 1, 2]])
  Set([[1, 3, 3, 3, 2, 2], [1, 3, 3, 2, 2, 2]])
  Set([[2, 3, 1, 1, 1, 3], [2, 3, 1, 1, 3, 3]])
  Set([[3, 3, 2, 1, 3, 2], [2, 3, 2, 1, 3, 2]

Set{Set{Vector{Int64}}} with 360 elements:
  Set([[2, 3, 1, 3, 3, 2], [3, 1, 2, 1, 1, 3]])
  Set([[3, 3, 2, 1, 2, 3], [1, 1, 3, 2, 3, 1]])
  Set([[1, 2, 1, 3, 1, 3], [3, 1, 3, 2, 3, 2]])
  Set([[3, 1, 2, 1, 2, 1], [2, 3, 1, 3, 1, 3]])
  Set([[3, 2, 2, 2, 1, 3], [1, 3, 3, 3, 2, 1]])
  Set([[2, 3, 3, 3, 1, 2], [3, 1, 1, 1, 2, 3]])
  Set([[3, 3, 3, 1, 2, 1], [1, 1, 1, 2, 3, 2]])
  Set([[1, 2, 1, 3, 3, 1], [2, 3, 2, 1, 1, 2]])
  Set([[3, 2, 3, 1, 3, 2], [1, 3, 1, 2, 1, 3]])
  Set([[3, 3, 1, 3, 2, 2], [1, 1, 2, 1, 3, 3]])
  Set([[3, 1, 2, 1, 1, 2], [1, 2, 3, 2, 2, 3]])
  Set([[3, 2, 1, 2, 1, 2], [1, 3, 2, 3, 2, 3]])
  Set([[1, 1, 2, 2, 3, 1], [2, 2, 3, 3, 1, 2]])
  Set([[1, 3, 2, 3, 3, 1], [2, 1, 3, 1, 1, 2]])
  Set([[2, 3, 1, 1, 2, 1], [3, 1, 2, 2, 3, 2]])
  Set([[2, 3, 1, 2, 3, 3], [3, 1, 2, 3, 1, 1]])
  Set([[2, 3, 2, 3, 3, 1], [3, 1, 3, 1, 1, 2]])
  Set([[3, 3, 2, 1, 2, 3], [2, 2, 1, 3, 1, 2]])
  Set([[3, 3, 2, 1, 1, 3], [1, 1, 3, 2, 2, 1]])
  Set([[3, 3, 1, 3, 2, 1], [2, 2, 3, 2, 1, 3]

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[1, 2, 3, 2, 2, 3], [2, 3, 2, 1, 3, 2]])
  Set([[1, 2, 3, 2, 1, 1], [2, 1, 1, 1, 2, 3]])
  Set([[3, 3, 2, 2, 1, 2], [2, 2, 3, 1, 2, 3]])
  Set([[2, 1, 3, 1, 1, 3], [1, 2, 1, 3, 3, 1]])
  Set([[2, 3, 3, 2, 1, 3], [3, 1, 2, 3, 3, 2]])
  Set([[2, 3, 3, 3, 1, 2], [3, 2, 2, 1, 3, 3]])
  Set([[2, 1, 2, 3, 3, 3], [3, 3, 3, 2, 2, 1]])
  Set([[3, 3, 1, 3, 2, 1], [2, 1, 3, 1, 3, 3]])
  Set([[3, 1, 2, 3, 1, 1], [1, 2, 1, 1, 3, 3]])
  Set([[2, 2, 2, 3, 3, 1], [1, 3, 3, 2, 2, 2]])
  Set([[1, 3, 2, 3, 2, 2], [2, 2, 1, 2, 3, 3]])
  Set([[3, 2, 3, 1, 3, 2], [2, 3, 2, 3, 1, 3]])
  Set([[3, 2, 1, 1, 3, 1], [1, 1, 2, 3, 1, 3]])
  Set([[1, 3, 1, 1, 3, 2], [2, 1, 3, 3, 1, 1]])
  Set([[2, 1, 1, 3, 3, 1], [1, 3, 3, 1, 1, 2]])
  Set([[1, 2, 3, 1, 2, 1], [2, 1, 1, 2, 1, 3]])
  Set([[1, 3, 3, 3, 1, 2], [3, 2, 1, 1, 3, 3]])
  Set([[1, 1, 3, 2, 2, 1], [2, 3, 1, 1, 1, 2]])
  Set([[1, 2, 3, 3, 2, 2], [2, 1, 2, 2, 3, 3]])
  Set([[3, 3, 1, 1, 3, 2], [2, 1, 3, 3, 1, 3]

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[2, 1, 2, 3, 3, 3], [2, 1, 3, 2, 2, 3]])
  Set([[1, 3, 3, 2, 2, 3], [1, 2, 3, 2, 3, 2]])
  Set([[3, 2, 3, 3, 2, 1], [3, 3, 2, 2, 2, 1]])
  Set([[1, 2, 1, 3, 1, 3], [3, 2, 3, 1, 1, 3]])
  Set([[3, 3, 1, 3, 2, 1], [1, 1, 1, 3, 2, 3]])
  Set([[1, 2, 3, 2, 1, 2], [2, 2, 3, 1, 1, 1]])
  Set([[1, 3, 3, 3, 2, 1], [1, 1, 1, 3, 2, 3]])
  Set([[2, 2, 3, 1, 2, 1], [2, 1, 3, 1, 1, 2]])
  Set([[1, 3, 2, 2, 3, 2], [1, 2, 3, 3, 3, 2]])
  Set([[3, 3, 2, 3, 1, 2], [2, 3, 3, 2, 1, 2]])
  Set([[3, 2, 3, 2, 2, 1], [3, 2, 2, 3, 3, 1]])
  Set([[1, 3, 2, 3, 3, 1], [1, 1, 2, 3, 1, 3]])
  Set([[1, 2, 2, 2, 1, 3], [2, 2, 1, 1, 1, 3]])
  Set([[3, 2, 2, 3, 1, 3], [2, 3, 2, 2, 1, 3]])
  Set([[3, 2, 3, 1, 2, 3], [2, 2, 2, 1, 3, 3]])
  Set([[2, 2, 2, 1, 3, 1], [1, 2, 1, 1, 3, 2]])
  Set([[1, 1, 3, 2, 1, 2], [2, 1, 3, 1, 2, 2]])
  Set([[3, 1, 2, 3, 2, 3], [2, 1, 2, 3, 3, 2]])
  Set([[3, 1, 2, 2, 3, 2], [3, 1, 3, 2, 2, 3]])
  Set([[2, 1, 2, 1, 3, 2], [1, 2, 1, 1, 3, 2

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[2, 3, 2, 1, 2, 3], [3, 1, 2, 3, 2, 2]])
  Set([[3, 3, 1, 2, 2, 3], [3, 2, 2, 3, 1, 3]])
  Set([[2, 3, 2, 3, 3, 1], [3, 3, 1, 2, 3, 2]])
  Set([[1, 2, 2, 3, 2, 3], [3, 3, 2, 2, 2, 1]])
  Set([[3, 2, 1, 2, 2, 3], [2, 2, 3, 3, 2, 1]])
  Set([[2, 2, 3, 3, 1, 3], [3, 1, 3, 2, 2, 3]])
  Set([[1, 1, 3, 3, 3, 2], [2, 3, 3, 1, 3, 1]])
  Set([[2, 2, 3, 2, 1, 1], [2, 1, 1, 2, 2, 3]])
  Set([[1, 1, 2, 3, 3, 1], [1, 1, 3, 2, 1, 3]])
  Set([[2, 1, 3, 1, 1, 2], [1, 2, 2, 1, 1, 3]])
  Set([[2, 2, 3, 1, 1, 2], [1, 2, 1, 3, 2, 2]])
  Set([[3, 3, 3, 1, 1, 2], [3, 1, 3, 3, 2, 1]])
  Set([[2, 1, 1, 2, 3, 1], [3, 1, 1, 1, 2, 2]])
  Set([[2, 2, 1, 3, 2, 3], [2, 3, 3, 1, 2, 2]])
  Set([[1, 2, 3, 1, 2, 2], [3, 2, 1, 2, 1, 2]])
  Set([[2, 3, 3, 2, 2, 1], [3, 2, 1, 2, 2, 3]])
  Set([[3, 3, 1, 3, 2, 2], [3, 2, 2, 3, 1, 3]])
  Set([[3, 3, 2, 2, 1, 2], [2, 1, 2, 2, 3, 3]])
  Set([[1, 2, 2, 3, 3, 2], [3, 3, 2, 1, 2, 2]])
  Set([[2, 1, 2, 3, 3, 3], [1, 2, 3, 2, 3, 3

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[1, 3, 3, 3, 2, 1], [1, 2, 1, 1, 3, 2]])
  Set([[2, 3, 3, 3, 1, 2], [1, 1, 3, 1, 2, 3]])
  Set([[3, 1, 3, 3, 1, 2], [2, 1, 1, 1, 2, 3]])
  Set([[3, 2, 2, 1, 1, 2], [2, 1, 1, 3, 1, 3]])
  Set([[1, 1, 2, 2, 3, 1], [2, 3, 2, 3, 1, 2]])
  Set([[3, 3, 1, 1, 2, 3], [2, 1, 2, 1, 3, 1]])
  Set([[3, 2, 1, 2, 2, 1], [1, 2, 3, 3, 3, 2]])
  Set([[1, 1, 3, 2, 3, 3], [1, 2, 1, 3, 1, 2]])
  Set([[3, 1, 3, 3, 2, 1], [2, 2, 3, 2, 1, 3]])
  Set([[2, 2, 1, 1, 3, 1], [1, 3, 1, 3, 2, 3]])
  Set([[1, 2, 1, 2, 3, 2], [1, 1, 3, 1, 2, 3]])
  Set([[3, 3, 1, 2, 2, 2], [1, 2, 3, 1, 2, 1]])
  Set([[3, 1, 2, 3, 1, 3], [2, 2, 3, 1, 1, 1]])
  Set([[1, 2, 1, 3, 1, 2], [3, 2, 2, 1, 2, 3]])
  Set([[2, 2, 3, 1, 1, 2], [3, 2, 1, 3, 2, 3]])
  Set([[1, 1, 1, 2, 3, 3], [1, 2, 2, 3, 1, 2]])
  Set([[2, 1, 3, 2, 3, 2], [3, 2, 1, 3, 3, 1]])
  Set([[1, 3, 2, 3, 2, 2], [2, 3, 3, 1, 3, 1]])
  Set([[2, 1, 3, 3, 1, 3], [3, 1, 2, 1, 2, 1]])
  Set([[2, 3, 3, 1, 1, 3], [1, 2, 2, 2, 3, 3

Set{Set{Vector{Int64}}} with 2160 elements:
  Set([[3, 2, 1, 1, 1, 2], [3, 3, 2, 1, 1, 1]])
  Set([[2, 1, 3, 2, 2, 3], [2, 2, 1, 1, 2, 3]])
  Set([[2, 1, 3, 2, 2, 1], [1, 3, 3, 2, 2, 2]])
  Set([[2, 2, 1, 3, 2, 1], [3, 2, 1, 2, 2, 3]])
  Set([[3, 3, 1, 2, 3, 2], [3, 3, 3, 2, 1, 1]])
  Set([[2, 1, 3, 2, 1, 1], [3, 2, 3, 1, 1, 1]])
  Set([[2, 3, 1, 2, 3, 3], [1, 3, 3, 2, 3, 1]])
  Set([[2, 3, 2, 1, 2, 3], [2, 1, 3, 1, 2, 2]])
  Set([[2, 1, 2, 1, 1, 3], [1, 2, 3, 1, 1, 3]])
  Set([[3, 3, 2, 2, 1, 3], [2, 3, 1, 3, 1, 3]])
  Set([[3, 1, 3, 2, 2, 3], [3, 3, 3, 2, 1, 1]])
  Set([[1, 3, 2, 1, 3, 1], [1, 1, 2, 3, 2, 1]])
  Set([[3, 1, 3, 2, 2, 2], [3, 2, 1, 1, 2, 2]])
  Set([[3, 3, 2, 3, 2, 1], [3, 3, 1, 1, 2, 3]])
  Set([[1, 2, 1, 3, 2, 2], [2, 2, 3, 3, 2, 1]])
  Set([[2, 3, 1, 3, 3, 2], [2, 3, 3, 3, 1, 1]])
  Set([[2, 2, 1, 1, 3, 1], [3, 1, 1, 1, 3, 2]])
  Set([[3, 2, 2, 1, 2, 1], [2, 3, 2, 1, 2, 3]])
  Set([[3, 1, 2, 3, 3, 1], [3, 2, 3, 2, 3, 1]])
  Set([[1, 1, 2, 1, 2, 3], [1, 2, 3, 1, 1, 3

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[1, 1, 3, 1, 2, 3], [1, 2, 3, 1, 3, 3]])
  Set([[1, 2, 1, 1, 3, 3], [2, 3, 1, 1, 3, 3]])
  Set([[3, 3, 1, 2, 3, 1], [2, 3, 1, 1, 3, 1]])
  Set([[3, 1, 1, 2, 3, 3], [2, 1, 1, 1, 3, 3]])
  Set([[2, 2, 3, 1, 3, 2], [1, 2, 3, 3, 3, 2]])
  Set([[3, 3, 1, 2, 2, 3], [1, 3, 2, 2, 2, 3]])
  Set([[3, 2, 1, 2, 1, 2], [1, 2, 1, 2, 1, 3]])
  Set([[3, 2, 2, 1, 2, 1], [1, 2, 2, 1, 3, 1]])
  Set([[2, 1, 2, 2, 3, 1], [3, 1, 2, 2, 1, 1]])
  Set([[2, 1, 1, 1, 3, 2], [2, 1, 3, 1, 2, 2]])
  Set([[2, 3, 1, 1, 2, 1], [2, 2, 3, 1, 2, 1]])
  Set([[3, 2, 3, 2, 2, 1], [3, 2, 3, 2, 1, 3]])
  Set([[2, 1, 1, 2, 3, 1], [2, 3, 1, 2, 2, 1]])
  Set([[1, 2, 1, 1, 3, 2], [1, 2, 1, 3, 2, 2]])
  Set([[2, 3, 2, 1, 2, 1], [2, 1, 2, 1, 3, 1]])
  Set([[1, 3, 3, 2, 1, 3], [1, 2, 3, 1, 1, 3]])
  Set([[2, 2, 3, 1, 1, 1], [2, 2, 2, 1, 1, 3]])
  Set([[3, 2, 1, 1, 2, 2], [1, 3, 1, 1, 2, 2]])
  Set([[1, 2, 3, 1, 2, 2], [1, 3, 1, 1, 2, 2]])
  Set([[3, 1, 1, 1, 2, 2], [2, 1, 1, 3, 2, 2]

Set{Set{Vector{Int64}}} with 720 elements:
  Set([[1, 3, 2, 2, 1, 2], [2, 3, 1, 1, 3, 1]])
  Set([[2, 3, 3, 3, 1, 2], [1, 2, 2, 2, 1, 3]])
  Set([[2, 3, 2, 1, 3, 3], [3, 2, 1, 1, 2, 2]])
  Set([[3, 3, 2, 1, 3, 1], [2, 2, 3, 3, 2, 1]])
  Set([[2, 2, 3, 3, 1, 2], [1, 1, 3, 2, 2, 1]])
  Set([[2, 2, 1, 3, 2, 1], [1, 1, 2, 3, 1, 3]])
  Set([[2, 1, 1, 2, 3, 1], [3, 2, 2, 1, 3, 2]])
  Set([[2, 3, 1, 1, 3, 3], [3, 2, 1, 3, 2, 2]])
  Set([[3, 2, 3, 1, 2, 2], [2, 3, 1, 1, 3, 3]])
  Set([[1, 1, 1, 3, 2, 2], [3, 3, 3, 1, 1, 2]])
  Set([[3, 3, 2, 1, 2, 3], [2, 2, 1, 1, 3, 2]])
  Set([[2, 3, 1, 1, 2, 1], [1, 1, 3, 3, 2, 3]])
  Set([[3, 2, 3, 3, 1, 2], [2, 1, 2, 2, 1, 3]])
  Set([[2, 3, 3, 1, 1, 3], [2, 1, 1, 3, 2, 1]])
  Set([[3, 1, 1, 1, 2, 2], [3, 2, 2, 2, 1, 3]])
  Set([[2, 1, 2, 1, 1, 3], [1, 2, 3, 2, 2, 3]])
  Set([[2, 2, 2, 1, 3, 1], [1, 1, 1, 3, 3, 2]])
  Set([[2, 3, 1, 1, 2, 2], [1, 3, 2, 3, 1, 1]])
  Set([[1, 3, 1, 2, 3, 1], [3, 2, 3, 2, 1, 3]])
  Set([[3, 2, 2, 3, 1, 3], [1, 3, 2, 1, 3, 1]

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[2, 1, 3, 2, 1, 1], [3, 2, 3, 3, 1, 2]])
  Set([[3, 3, 3, 1, 2, 1], [3, 1, 1, 2, 2, 2]])
  Set([[2, 3, 2, 2, 1, 3], [2, 1, 3, 3, 1, 1]])
  Set([[2, 2, 1, 3, 2, 3], [3, 2, 1, 1, 3, 1]])
  Set([[3, 1, 2, 2, 2, 1], [3, 3, 1, 1, 2, 3]])
  Set([[3, 3, 1, 3, 2, 2], [3, 2, 1, 2, 1, 1]])
  Set([[1, 1, 2, 3, 1, 2], [2, 1, 3, 3, 2, 3]])
  Set([[3, 2, 1, 3, 2, 3], [3, 1, 1, 2, 1, 2]])
  Set([[3, 1, 2, 3, 3, 2], [2, 1, 1, 3, 2, 1]])
  Set([[2, 1, 3, 1, 2, 2], [1, 3, 3, 3, 2, 1]])
  Set([[2, 3, 1, 1, 1, 3], [2, 2, 1, 3, 3, 2]])
  Set([[1, 2, 3, 1, 2, 1], [2, 3, 3, 2, 3, 1]])
  Set([[1, 1, 3, 2, 2, 2], [3, 3, 3, 2, 1, 1]])
  Set([[2, 3, 1, 3, 2, 2], [2, 1, 1, 1, 3, 3]])
  Set([[1, 2, 3, 3, 2, 3], [1, 1, 3, 2, 1, 2]])
  Set([[2, 3, 2, 1, 3, 3], [1, 2, 1, 1, 2, 3]])
  Set([[1, 2, 1, 2, 1, 3], [2, 3, 2, 3, 1, 3]])
  Set([[3, 2, 3, 1, 3, 1], [1, 2, 3, 2, 1, 2]])
  Set([[3, 3, 2, 3, 1, 1], [1, 3, 2, 1, 2, 2]])
  Set([[1, 1, 1, 3, 3, 2], [1, 3, 3, 2, 2, 2]

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[2, 1, 2, 3, 1, 2], [1, 1, 1, 2, 3, 2]])
  Set([[2, 1, 3, 3, 2, 3], [2, 3, 2, 2, 1, 3]])
  Set([[3, 1, 3, 1, 3, 2], [1, 1, 3, 2, 1, 3]])
  Set([[1, 2, 2, 2, 1, 3], [1, 1, 1, 2, 3, 2]])
  Set([[3, 2, 1, 2, 2, 1], [2, 2, 1, 1, 1, 3]])
  Set([[3, 2, 3, 1, 2, 3], [3, 2, 2, 3, 1, 2]])
  Set([[1, 2, 1, 2, 3, 1], [2, 2, 1, 3, 1, 2]])
  Set([[3, 3, 1, 2, 1, 1], [3, 2, 1, 1, 3, 3]])
  Set([[3, 3, 2, 3, 2, 1], [3, 2, 1, 2, 2, 3]])
  Set([[1, 3, 2, 1, 3, 3], [1, 3, 3, 2, 1, 1]])
  Set([[3, 2, 3, 2, 1, 3], [2, 1, 2, 2, 3, 3]])
  Set([[1, 1, 1, 2, 3, 3], [3, 3, 1, 1, 3, 2]])
  Set([[1, 2, 2, 2, 3, 3], [2, 3, 3, 2, 3, 1]])
  Set([[3, 2, 2, 3, 1, 3], [2, 2, 1, 3, 3, 2]])
  Set([[2, 3, 1, 3, 2, 2], [2, 3, 2, 1, 3, 3]])
  Set([[3, 3, 3, 1, 2, 1], [1, 1, 3, 2, 3, 1]])
  Set([[1, 2, 2, 3, 2, 3], [2, 3, 3, 3, 2, 1]])
  Set([[2, 3, 1, 3, 3, 1], [3, 3, 1, 1, 1, 2]])
  Set([[2, 3, 1, 2, 3, 3], [1, 2, 3, 2, 2, 3]])
  Set([[2, 1, 3, 1, 2, 1], [3, 1, 1, 2, 2, 2

Set{Set{Vector{Int64}}} with 360 elements:
  Set([[1, 3, 1, 1, 3, 2], [1, 3, 1, 1, 2, 3]])
  Set([[2, 2, 1, 3, 2, 1], [2, 2, 1, 1, 2, 3]])
  Set([[1, 2, 3, 2, 1, 1], [1, 2, 2, 3, 1, 1]])
  Set([[1, 3, 3, 1, 1, 2], [1, 2, 3, 1, 1, 3]])
  Set([[2, 3, 1, 3, 2, 2], [2, 3, 3, 1, 2, 2]])
  Set([[3, 1, 1, 2, 2, 1], [2, 1, 1, 3, 2, 1]])
  Set([[3, 3, 1, 3, 2, 1], [3, 3, 2, 3, 1, 1]])
  Set([[3, 1, 3, 3, 2, 2], [3, 2, 3, 3, 1, 2]])
  Set([[2, 2, 2, 3, 1, 3], [2, 2, 2, 1, 3, 3]])
  Set([[1, 3, 1, 3, 1, 2], [1, 3, 1, 2, 1, 3]])
  Set([[2, 3, 2, 3, 2, 1], [2, 1, 2, 3, 2, 3]])
  Set([[3, 2, 2, 3, 3, 1], [3, 1, 2, 3, 3, 2]])
  Set([[2, 1, 3, 1, 1, 3], [3, 1, 2, 1, 1, 3]])
  Set([[2, 3, 2, 2, 1, 3], [2, 1, 2, 2, 3, 3]])
  Set([[3, 2, 1, 2, 1, 2], [1, 2, 3, 2, 1, 2]])
  Set([[3, 1, 3, 1, 3, 2], [3, 1, 3, 2, 3, 1]])
  Set([[3, 2, 3, 1, 2, 2], [3, 2, 1, 3, 2, 2]])
  Set([[1, 3, 2, 3, 3, 1], [2, 3, 1, 3, 3, 1]])
  Set([[1, 2, 3, 2, 2, 1], [1, 2, 1, 2, 2, 3]])
  Set([[3, 2, 1, 3, 1, 1], [2, 3, 1, 3, 1, 1]

Set{Set{Vector{Int64}}} with 360 elements:
  Set([[2, 3, 1, 3, 2, 3]])
  Set([[3, 1, 2, 3, 1, 1]])
  Set([[2, 3, 1, 2, 3, 2]])
  Set([[1, 1, 2, 3, 1, 3]])
  Set([[3, 3, 1, 2, 3, 2]])
  Set([[1, 2, 3, 2, 1, 2]])
  Set([[1, 1, 1, 3, 2, 3]])
  Set([[1, 3, 2, 2, 3, 2]])
  Set([[3, 1, 2, 1, 1, 2]])
  Set([[2, 1, 1, 2, 3, 2]])
  Set([[1, 3, 2, 3, 3, 2]])
  Set([[2, 1, 1, 1, 2, 3]])
  Set([[2, 2, 1, 1, 3, 1]])
  Set([[2, 2, 3, 3, 2, 1]])
  Set([[3, 1, 2, 1, 3, 3]])
  Set([[3, 3, 3, 2, 1, 2]])
  Set([[3, 1, 3, 3, 1, 2]])
  Set([[1, 2, 3, 2, 3, 3]])
  Set([[3, 2, 1, 2, 3, 2]])
  Set([[1, 2, 1, 1, 2, 3]])
  Set([[1, 2, 3, 1, 1, 3]])
  Set([[3, 1, 3, 1, 2, 1]])
  Set([[1, 1, 3, 1, 2, 3]])
  Set([[2, 1, 1, 1, 3, 3]])
  Set([[3, 1, 3, 2, 1, 3]])
  ⋮ 

Set{Set{Vector{Int64}}} with 180 elements:
  Set([[2, 2, 2, 1, 3, 1], [2, 2, 2, 3, 1, 3]])
  Set([[1, 2, 3, 3, 2, 3], [2, 1, 3, 3, 1, 3]])
  Set([[1, 3, 2, 2, 1, 2], [3, 1, 2, 2, 3, 2]])
  Set([[2, 1, 2, 3, 3, 3], [1, 2, 1, 3, 3, 3]])
  Set([[1, 1, 3, 2, 1, 2], [1, 1, 2, 3, 1, 3]])
  Set([[2, 3, 3, 2, 3, 1], [1, 3, 3, 1, 3, 2]])
  Set([[3, 3, 2, 1, 3, 2], [3, 3, 1, 2, 3, 1]])
  Set([[2, 1, 2, 1, 1, 3], [3, 1, 3, 1, 1, 2]])
  Set([[3, 1, 2, 2, 1, 1], [2, 1, 3, 3, 1, 1]])
  Set([[2, 3, 2, 3, 1, 3], [1, 3, 1, 3, 2, 3]])
  Set([[2, 3, 2, 3, 3, 1], [1, 3, 1, 3, 3, 2]])
  Set([[1, 1, 1, 2, 3, 2], [1, 1, 1, 3, 2, 3]])
  Set([[2, 3, 1, 1, 2, 1], [3, 2, 1, 1, 3, 1]])
  Set([[3, 3, 1, 2, 3, 2], [3, 3, 2, 1, 3, 1]])
  Set([[3, 2, 1, 3, 1, 1], [2, 3, 1, 2, 1, 1]])
  Set([[2, 2, 3, 3, 3, 1], [1, 1, 3, 3, 3, 2]])
  Set([[3, 3, 2, 2, 1, 2], [1, 1, 2, 2, 3, 2]])
  Set([[2, 1, 2, 1, 2, 3], [2, 3, 2, 3, 2, 1]])
  Set([[3, 3, 2, 2, 1, 3], [3, 3, 1, 1, 2, 3]])
  Set([[3, 1, 2, 1, 2, 2], [1, 3, 2, 3, 2, 2]

Set{Set{Vector{Int64}}} with 180 elements:
  Set([[2, 1, 2, 2, 1, 3], [1, 2, 1, 1, 2, 3]])
  Set([[3, 1, 1, 2, 2, 1], [3, 2, 2, 1, 1, 2]])
  Set([[3, 1, 2, 1, 1, 2], [3, 2, 1, 2, 2, 1]])
  Set([[3, 2, 2, 1, 1, 1], [3, 1, 1, 2, 2, 2]])
  Set([[2, 2, 1, 3, 2, 1], [1, 1, 2, 3, 1, 2]])
  Set([[1, 3, 1, 2, 3, 1], [3, 1, 3, 2, 1, 3]])
  Set([[3, 2, 1, 1, 1, 3], [1, 2, 3, 3, 3, 1]])
  Set([[1, 2, 3, 3, 2, 3], [1, 3, 2, 2, 3, 2]])
  Set([[1, 3, 3, 2, 2, 3], [1, 2, 2, 3, 3, 2]])
  Set([[2, 1, 3, 1, 3, 3], [2, 3, 1, 3, 1, 1]])
  Set([[3, 2, 2, 3, 2, 1], [2, 3, 3, 2, 3, 1]])
  Set([[3, 2, 1, 3, 1, 3], [1, 2, 3, 1, 3, 1]])
  Set([[3, 3, 1, 3, 1, 2], [1, 1, 3, 1, 3, 2]])
  Set([[2, 1, 3, 1, 2, 1], [1, 2, 3, 2, 1, 2]])
  Set([[3, 1, 3, 1, 3, 2], [1, 3, 1, 3, 1, 2]])
  Set([[3, 1, 1, 2, 1, 3], [1, 3, 3, 2, 3, 1]])
  Set([[1, 2, 3, 1, 1, 3], [3, 2, 1, 3, 3, 1]])
  Set([[3, 2, 3, 2, 1, 2], [2, 3, 2, 3, 1, 3]])
  Set([[3, 2, 1, 2, 3, 2], [2, 3, 1, 3, 2, 3]])
  Set([[2, 2, 2, 3, 1, 1], [1, 1, 1, 3, 2, 2]

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[2, 2, 3, 1, 3, 2], [1, 1, 3, 1, 3, 2]])
  Set([[3, 2, 3, 2, 2, 1], [3, 1, 3, 1, 2, 1]])
  Set([[1, 1, 3, 2, 1, 3], [2, 2, 3, 2, 1, 3]])
  Set([[1, 2, 2, 1, 3, 2], [1, 3, 2, 1, 3, 3]])
  Set([[1, 2, 3, 2, 1, 2], [1, 3, 3, 2, 1, 3]])
  Set([[3, 3, 1, 1, 2, 1], [3, 3, 2, 2, 2, 1]])
  Set([[3, 3, 3, 1, 2, 2], [1, 3, 1, 1, 2, 2]])
  Set([[2, 1, 3, 2, 3, 3], [2, 1, 1, 2, 1, 3]])
  Set([[2, 1, 2, 2, 3, 1], [2, 1, 3, 3, 3, 1]])
  Set([[3, 3, 2, 3, 2, 1], [3, 1, 2, 1, 2, 1]])
  Set([[1, 1, 1, 2, 3, 3], [2, 2, 1, 2, 3, 3]])
  Set([[3, 2, 2, 3, 1, 3], [1, 2, 2, 1, 1, 3]])
  Set([[3, 2, 1, 1, 3, 1], [3, 2, 2, 1, 3, 2]])
  Set([[3, 2, 3, 1, 3, 2], [1, 2, 1, 1, 3, 2]])
  Set([[2, 2, 1, 3, 3, 3], [2, 2, 1, 1, 1, 3]])
  Set([[3, 2, 2, 2, 1, 1], [3, 3, 3, 2, 1, 1]])
  Set([[1, 1, 3, 3, 2, 3], [1, 1, 2, 2, 2, 3]])
  Set([[3, 1, 2, 1, 1, 3], [3, 2, 2, 1, 2, 3]])
  Set([[1, 1, 2, 3, 2, 2], [1, 1, 3, 3, 2, 3]])
  Set([[3, 2, 2, 1, 1, 2], [3, 2, 3, 1, 1, 3]

Set{Set{Vector{Int64}}} with 1080 elements:
  Set([[3, 2, 2, 2, 3, 1], [1, 1, 2, 2, 2, 3]])
  Set([[3, 3, 1, 2, 3, 1], [3, 2, 3, 1, 3, 2]])
  Set([[3, 1, 2, 3, 1, 3], [2, 2, 1, 3, 3, 3]])
  Set([[2, 2, 3, 1, 3, 2], [1, 2, 1, 3, 2, 2]])
  Set([[3, 2, 1, 1, 2, 1], [2, 3, 1, 1, 1, 3]])
  Set([[2, 2, 3, 1, 1, 2], [2, 2, 1, 2, 3, 3]])
  Set([[3, 3, 1, 2, 1, 1], [2, 1, 1, 3, 1, 2]])
  Set([[2, 1, 3, 2, 3, 2], [2, 3, 1, 2, 2, 1]])
  Set([[2, 1, 3, 1, 2, 1], [3, 1, 2, 3, 1, 1]])
  Set([[1, 2, 3, 2, 1, 1], [3, 1, 2, 3, 1, 1]])
  Set([[2, 3, 2, 1, 1, 1], [1, 2, 3, 1, 1, 3]])
  Set([[2, 2, 3, 2, 1, 1], [2, 2, 1, 3, 3, 2]])
  Set([[1, 1, 3, 2, 2, 1], [1, 1, 2, 1, 3, 3]])
  Set([[3, 1, 1, 2, 1, 3], [1, 1, 1, 3, 2, 2]])
  Set([[3, 1, 2, 1, 1, 2], [2, 1, 3, 3, 1, 1]])
  Set([[3, 2, 3, 1, 2, 2], [2, 2, 1, 3, 2, 1]])
  Set([[2, 2, 3, 3, 3, 1], [1, 3, 3, 1, 3, 2]])
  Set([[3, 2, 1, 1, 3, 3], [3, 1, 3, 2, 3, 2]])
  Set([[2, 2, 1, 3, 1, 1], [1, 3, 3, 2, 1, 1]])
  Set([[1, 2, 2, 2, 3, 1], [3, 2, 3, 2, 1, 2

Set{Set{Vector{Int64}}} with 540 elements:
  Set([[3, 1, 3, 2, 3, 1], [3, 3, 2, 1, 2, 3]])
  Set([[3, 1, 1, 1, 3, 2], [1, 1, 2, 2, 1, 3]])
  Set([[1, 2, 1, 3, 2, 1], [1, 1, 3, 2, 1, 3]])
  Set([[3, 1, 1, 2, 2, 2], [1, 2, 2, 3, 3, 2]])
  Set([[1, 2, 1, 3, 1, 2], [3, 1, 3, 2, 1, 1]])
  Set([[3, 2, 2, 1, 2, 1], [1, 3, 3, 2, 2, 2]])
  Set([[1, 3, 2, 1, 2, 1], [3, 2, 1, 3, 1, 1]])
  Set([[3, 2, 1, 2, 1, 1], [2, 1, 1, 1, 3, 3]])
  Set([[2, 3, 3, 1, 2, 3], [3, 1, 3, 2, 3, 1]])
  Set([[3, 2, 2, 1, 1, 1], [2, 1, 1, 3, 1, 3]])
  Set([[3, 2, 2, 3, 2, 1], [2, 1, 2, 2, 1, 3]])
  Set([[3, 1, 3, 2, 3, 2], [3, 2, 1, 3, 1, 3]])
  Set([[3, 2, 1, 1, 3, 1], [1, 3, 2, 1, 1, 2]])
  Set([[2, 3, 1, 1, 1, 2], [1, 2, 3, 1, 3, 1]])
  Set([[1, 2, 2, 3, 2, 1], [2, 3, 2, 1, 3, 2]])
  Set([[2, 2, 1, 1, 2, 3], [3, 3, 2, 2, 2, 1]])
  Set([[1, 3, 2, 3, 2, 2], [3, 2, 2, 2, 1, 1]])
  Set([[2, 2, 1, 1, 3, 1], [1, 1, 1, 3, 2, 3]])
  Set([[3, 1, 2, 1, 1, 2], [2, 1, 1, 3, 3, 1]])
  Set([[2, 3, 1, 3, 3, 2], [3, 1, 2, 1, 3, 3]

Set{Set{Vector{Int64}}} with 360 elements:
  Set([[1, 1, 2, 3, 2, 2], [1, 2, 3, 1, 3, 3]])
  Set([[2, 2, 3, 3, 3, 1], [2, 3, 1, 1, 1, 2]])
  Set([[3, 2, 1, 1, 3, 1], [1, 3, 2, 2, 3, 2]])
  Set([[3, 3, 2, 2, 1, 3], [1, 1, 2, 3, 2, 1]])
  Set([[2, 1, 3, 2, 2, 3], [1, 3, 3, 1, 1, 2]])
  Set([[1, 1, 2, 1, 2, 3], [3, 3, 1, 3, 2, 2]])
  Set([[1, 2, 1, 1, 3, 3], [2, 3, 2, 2, 1, 3]])
  Set([[3, 1, 2, 1, 2, 2], [1, 1, 3, 2, 3, 3]])
  Set([[3, 2, 1, 1, 1, 3], [3, 3, 2, 2, 2, 1]])
  Set([[1, 2, 1, 2, 1, 3], [3, 2, 3, 1, 3, 2]])
  Set([[1, 3, 3, 3, 2, 2], [2, 1, 1, 1, 2, 3]])
  Set([[3, 2, 1, 2, 3, 2], [2, 1, 3, 1, 3, 1]])
  Set([[2, 1, 2, 3, 2, 3], [1, 3, 1, 2, 1, 3]])
  Set([[1, 1, 2, 2, 2, 3], [2, 1, 3, 3, 3, 1]])
  Set([[2, 1, 1, 3, 2, 2], [3, 2, 1, 1, 3, 3]])
  Set([[3, 2, 1, 3, 2, 2], [3, 1, 3, 2, 1, 1]])
  Set([[2, 2, 3, 1, 3, 2], [1, 1, 3, 3, 2, 1]])
  Set([[2, 3, 1, 2, 2, 1], [3, 1, 1, 3, 3, 2]])
  Set([[1, 3, 2, 2, 1, 2], [1, 1, 3, 3, 2, 3]])
  Set([[3, 2, 2, 2, 1, 3], [2, 1, 1, 1, 3, 3]

Set{Set{Vector{Int64}}} with 180 elements:
  Set([[2, 1, 2, 3, 3, 2], [1, 2, 1, 3, 3, 1]])
  Set([[3, 2, 2, 3, 2, 1], [3, 1, 1, 3, 1, 2]])
  Set([[3, 2, 1, 1, 1, 2], [1, 2, 3, 3, 3, 2]])
  Set([[3, 2, 1, 1, 3, 1], [3, 1, 2, 2, 3, 2]])
  Set([[3, 2, 2, 1, 3, 3], [1, 2, 2, 3, 1, 1]])
  Set([[2, 3, 3, 1, 2, 2], [1, 3, 3, 2, 1, 1]])
  Set([[1, 2, 2, 3, 2, 3], [2, 1, 1, 3, 1, 3]])
  Set([[2, 2, 3, 1, 1, 2], [3, 3, 2, 1, 1, 3]])
  Set([[3, 2, 3, 1, 3, 2], [1, 2, 1, 3, 1, 2]])
  Set([[2, 2, 1, 3, 2, 1], [3, 3, 1, 2, 3, 1]])
  Set([[2, 3, 2, 1, 3, 3], [2, 1, 2, 3, 1, 1]])
  Set([[3, 3, 1, 1, 1, 2], [3, 3, 2, 2, 2, 1]])
  Set([[3, 2, 3, 1, 2, 2], [3, 1, 3, 2, 1, 1]])
  Set([[1, 2, 1, 3, 2, 1], [3, 2, 3, 1, 2, 3]])
  Set([[1, 2, 3, 2, 1, 2], [1, 3, 2, 3, 1, 3]])
  Set([[1, 1, 1, 2, 3, 3], [2, 2, 2, 1, 3, 3]])
  Set([[1, 3, 3, 3, 1, 2], [1, 2, 2, 2, 1, 3]])
  Set([[1, 2, 2, 3, 2, 1], [1, 3, 3, 2, 3, 1]])
  Set([[2, 3, 2, 3, 1, 2], [1, 3, 1, 3, 2, 1]])
  Set([[3, 1, 1, 3, 2, 1], [3, 2, 2, 3, 1, 2]

Set{Set{Vector{Int64}}} with 360 elements:
  Set([[3, 3, 2, 2, 1, 2], [2, 1, 3, 3, 2, 3]])
  Set([[2, 3, 3, 1, 1, 3], [3, 1, 1, 3, 2, 1]])
  Set([[2, 1, 3, 1, 2, 2], [1, 3, 2, 2, 1, 1]])
  Set([[3, 3, 3, 2, 1, 2], [2, 2, 2, 3, 3, 1]])
  Set([[1, 3, 3, 1, 2, 3], [2, 1, 1, 3, 3, 1]])
  Set([[2, 1, 3, 1, 3, 1], [1, 3, 1, 3, 2, 3]])
  Set([[2, 1, 1, 1, 3, 2], [1, 2, 2, 2, 1, 3]])
  Set([[2, 3, 2, 3, 3, 1], [1, 2, 3, 2, 2, 3]])
  Set([[1, 2, 2, 3, 2, 3], [2, 3, 3, 1, 3, 2]])
  Set([[2, 1, 3, 3, 1, 3], [3, 3, 1, 1, 2, 1]])
  Set([[1, 2, 1, 3, 2, 1], [2, 3, 2, 1, 1, 2]])
  Set([[2, 2, 1, 2, 3, 1], [1, 1, 3, 1, 2, 2]])
  Set([[3, 1, 2, 3, 1, 3], [1, 3, 3, 1, 2, 1]])
  Set([[3, 3, 1, 1, 3, 2], [1, 1, 3, 2, 1, 3]])
  Set([[3, 1, 1, 2, 1, 2], [1, 2, 2, 1, 2, 3]])
  Set([[3, 1, 1, 1, 3, 2], [2, 3, 3, 3, 1, 1]])
  Set([[1, 3, 2, 3, 2, 2], [2, 2, 3, 1, 3, 3]])
  Set([[2, 2, 1, 3, 1, 1], [1, 3, 2, 1, 2, 2]])
  Set([[2, 2, 3, 2, 3, 1], [3, 3, 2, 3, 1, 2]])
  Set([[2, 2, 1, 3, 2, 1], [1, 1, 2, 2, 1, 3]

In [174]:
number_of_groups_with_size = [(i, Base.count(x -> length(x) == i, groups2)) for i in unique(map(x -> length(x), groups2))]

7-element Vector{Tuple{Int64, Int64}}:
 (2160, 18)
 (540, 12)
 (1080, 11)
 (4320, 1)
 (360, 6)
 (720, 1)
 (180, 3)

In [175]:
mapreduce(((size, number),) -> size * number, +, number_of_groups_with_size)

64980

???? This is 259920 / 64980 = 4 times more than expected!! (Fixed, did permutations(1:4) instead of permutations(1:3) lol)

In [176]:
mapreduce(((size, number),) -> number, +, number_of_groups_with_size)

52

Ok, that's expected...

In [132]:
number_of_groups_with_size = [(i, Base.count(x -> length(x) == i, groups1)) for i in unique(map(x -> length(x), groups1))]

8-element Vector{Tuple{Int64, Int64}}:
 (480, 1)
 (72, 1)
 (1080, 1)
 (1440, 1)
 (360, 2)
 (4, 1)
 (120, 1)
 (180, 1)

In [133]:
mapreduce(((size, number),) -> size * number, +, number_of_groups_with_size)

4096

Ok, this also make sense...

In [147]:
isempty(intersect(groups2[1], groups2[3]))

true